<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Analysis protocol | scRNA-seq data analysis</title>
  <meta name="description" content="This tutorial contains steps for analysing scRNA-seq data." />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Analysis protocol | scRNA-seq data analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This tutorial contains steps for analysing scRNA-seq data." />
  <meta name="github-repo" content="chisangad/scRNAseq" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Analysis protocol | scRNA-seq data analysis" />
  
  <meta name="twitter:description" content="This tutorial contains steps for analysing scRNA-seq data." />
  

<meta name="author" content="David Chisanga and Wei Shi" />


<meta name="date" content="2021-09-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="quantification-1.html"/>
<link rel="next" href="references.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://www.onjcri.org.au/our-research/#bcgl">BCG Lab @ONJCRI</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="intro.html"><a href="intro.html#quantification"><i class="fa fa-check"></i><b>2.1</b> Quantification</a></li>
<li class="chapter" data-level="2.2" data-path="intro.html"><a href="intro.html#quality-control-and-normalization"><i class="fa fa-check"></i><b>2.2</b> Quality control and Normalization</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="intro.html"><a href="intro.html#quality-control"><i class="fa fa-check"></i><b>2.2.1</b> Quality control</a></li>
<li class="chapter" data-level="2.2.2" data-path="intro.html"><a href="intro.html#normalisation-and-data-correction"><i class="fa fa-check"></i><b>2.2.2</b> Normalisation and data correction</a></li>
<li class="chapter" data-level="2.2.3" data-path="intro.html"><a href="intro.html#dimension-reduction"><i class="fa fa-check"></i><b>2.2.3</b> Dimension reduction</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="intro.html"><a href="intro.html#data-analysis"><i class="fa fa-check"></i><b>2.3</b> Data analysis</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>3</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="3.1" data-path="prerequisites.html"><a href="prerequisites.html#prereq"><i class="fa fa-check"></i><b>3.1</b> Data</a></li>
<li class="chapter" data-level="3.2" data-path="prerequisites.html"><a href="prerequisites.html#software"><i class="fa fa-check"></i><b>3.2</b> SOFTWARE</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="running-environment.html"><a href="running-environment.html"><i class="fa fa-check"></i><b>4</b> Running Environment</a>
<ul>
<li class="chapter" data-level="4.1" data-path="running-environment.html"><a href="running-environment.html#rstudio"><i class="fa fa-check"></i><b>4.1</b> RStudio</a></li>
<li class="chapter" data-level="4.2" data-path="running-environment.html"><a href="running-environment.html#setup"><i class="fa fa-check"></i><b>4.2</b> UNIX server + laptop</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quantification-1.html"><a href="quantification-1.html"><i class="fa fa-check"></i><b>5</b> Quantification</a>
<ul>
<li class="chapter" data-level="5.1" data-path="quantification-1.html"><a href="quantification-1.html#build-index-for-a-reference-genome"><i class="fa fa-check"></i><b>5.1</b> Build index for a reference genome</a></li>
<li class="chapter" data-level="5.2" data-path="quantification-1.html"><a href="quantification-1.html#counts"><i class="fa fa-check"></i><b>5.2</b> Map and quantify single cell RNA-seq data</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="analysis-protocol.html"><a href="analysis-protocol.html"><i class="fa fa-check"></i><b>6</b> Analysis protocol</a>
<ul>
<li class="chapter" data-level="6.1" data-path="analysis-protocol.html"><a href="analysis-protocol.html#read-mapping-and-umi-counts"><i class="fa fa-check"></i><b>6.1</b> Read mapping and UMI counts</a></li>
<li class="chapter" data-level="6.2" data-path="analysis-protocol.html"><a href="analysis-protocol.html#quality-control-checks"><i class="fa fa-check"></i><b>6.2</b> Quality control checks</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="analysis-protocol.html"><a href="analysis-protocol.html#mitochondria-content"><i class="fa fa-check"></i><b>6.2.1</b> Mitochondria content</a></li>
<li class="chapter" data-level="6.2.2" data-path="analysis-protocol.html"><a href="analysis-protocol.html#ribosomal-content"><i class="fa fa-check"></i><b>6.2.2</b> Ribosomal content</a></li>
<li class="chapter" data-level="6.2.3" data-path="analysis-protocol.html"><a href="analysis-protocol.html#number-of-detected-genes"><i class="fa fa-check"></i><b>6.2.3</b> Number of detected genes</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="analysis-protocol.html"><a href="analysis-protocol.html#filtering"><i class="fa fa-check"></i><b>6.3</b> Filtering</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="analysis-protocol.html"><a href="analysis-protocol.html#cell-filtering"><i class="fa fa-check"></i><b>6.3.1</b> Cell filtering</a></li>
<li class="chapter" data-level="6.3.2" data-path="analysis-protocol.html"><a href="analysis-protocol.html#gene-filtering"><i class="fa fa-check"></i><b>6.3.2</b> Gene filtering</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="analysis-protocol.html"><a href="analysis-protocol.html#normalization"><i class="fa fa-check"></i><b>6.4</b> Normalization</a></li>
<li class="chapter" data-level="6.5" data-path="analysis-protocol.html"><a href="analysis-protocol.html#dimension-reduction-and-clustering"><i class="fa fa-check"></i><b>6.5</b> Dimension reduction and clustering</a></li>
<li class="chapter" data-level="6.6" data-path="analysis-protocol.html"><a href="analysis-protocol.html#cell-annotation"><i class="fa fa-check"></i><b>6.6</b> Cell annotation</a></li>
<li class="chapter" data-level="6.7" data-path="analysis-protocol.html"><a href="analysis-protocol.html#trajectory-analysis"><i class="fa fa-check"></i><b>6.7</b> Trajectory analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">scRNA-seq data analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="analysis-protocol" class="section level1" number="6">
<h1><span class="header-section-number">Chapter 6</span> Analysis protocol</h1>
<div id="read-mapping-and-umi-counts" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Read mapping and UMI counts</h2>
<p>As discussed in section <a href="quantification-1.html#counts">5.2</a>, de-multiplexing, alignment to the GRCh38 transcriptome and unique molecular identifier (UMI)-collapsing were performed using <em>cellCounts</em>. Here we load the saved counts R object.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="analysis-protocol.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;counts.RData&quot;</span>)</span>
<span id="cb15-2"><a href="analysis-protocol.html#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(counts)</span></code></pre></div>
<pre><code>## [1] &quot;counts&quot;          &quot;cell.confidence&quot; &quot;annotation&quot;      &quot;sample.info&quot;</code></pre>
<p>The information below provides a summary of the quantification stats returned by <em>cellCounts</em>.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="analysis-protocol.html#cb17-1" aria-hidden="true" tabindex="-1"></a>counts<span class="sc">$</span>sample.info</span></code></pre></div>
<pre><code>##    SampleName TotalCells HighConfidenceCells RescuedCells TotalUMI MinUMI
## 7  LLU_A_Seq1       2954                2691          263 59485648    506
## 12 LLU_B_Seq1       1451                1305          146 33887743    504
##    MedianUMI MaxUMI  MeanUMI TotalReads MappedReads AssignedReads
## 7      18718  91172 20137.32  132228861   106079174      83649022
## 12     22124  99487 23354.75  134624084   105735336      78858247</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="analysis-protocol.html#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co">#rename sample name</span></span></code></pre></div>
<p>In addition, the sample information discussed in <a href="prerequisites.html#prereq">3.1</a> is loaded as shown by the details below.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="analysis-protocol.html#cb20-1" aria-hidden="true" tabindex="-1"></a>sample.info <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;sample_info.csv&quot;</span>,<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb20-2"><a href="analysis-protocol.html#cb20-2" aria-hidden="true" tabindex="-1"></a>sample.info</span></code></pre></div>
<pre><code>##   Library.Name                                 cell_line Sample.Name    sex
## 1   LLU_A_Seq1         breast cancer cell line (HCC1395)     HCC1395 female
## 2   LLU_B_Seq1 normal B lymphocyte cell line (HCC1395BL)   HCC1395BL female
##   tissue
## 1 breast
## 2  blood</code></pre>
<p>The counts are transformed to a <strong>Seurat</strong> object using the <em>CreateSeuratObject</em> function. Entrez gene IDs are also converted to their corresponding gene symbols.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="analysis-protocol.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co">#load Seurat package</span></span>
<span id="cb22-2"><a href="analysis-protocol.html#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb22-3"><a href="analysis-protocol.html#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="co">#load annotation package </span></span>
<span id="cb22-4"><a href="analysis-protocol.html#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb22-5"><a href="analysis-protocol.html#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Generate a list mapping gene Ids to symbols</span></span>
<span id="cb22-6"><a href="analysis-protocol.html#cb22-6" aria-hidden="true" tabindex="-1"></a>ID2symbol.lst<span class="ot">&lt;-</span><span class="fu">as.list</span>(org.Hs.egSYMBOL)</span>
<span id="cb22-7"><a href="analysis-protocol.html#cb22-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">rownames</span>(counts<span class="sc">$</span>counts<span class="sc">$</span>LLU_A_Seq1))</span></code></pre></div>
<pre><code>## [1] &quot;100287102&quot; &quot;653635&quot;    &quot;102466751&quot; &quot;100302278&quot; &quot;645520&quot;    &quot;79501&quot;</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="analysis-protocol.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Get list of Mitochondrial genes to be used for calculating percentage </span></span>
<span id="cb24-2"><a href="analysis-protocol.html#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="co">#of UMIs assigned to Mitochondrial genes in each cell</span></span>
<span id="cb24-3"><a href="analysis-protocol.html#cb24-3" aria-hidden="true" tabindex="-1"></a>genes.MT <span class="ot">&lt;-</span> <span class="fu">unlist</span>(ID2symbol.lst[<span class="fu">as.list</span>(<span class="fu">revmap</span>(org.Hs.egCHR))<span class="sc">$</span>MT],</span>
<span id="cb24-4"><a href="analysis-protocol.html#cb24-4" aria-hidden="true" tabindex="-1"></a>                   <span class="at">use.names =</span> F)</span></code></pre></div>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="analysis-protocol.html#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Create list of Seurat objects</span></span>
<span id="cb25-2"><a href="analysis-protocol.html#cb25-2" aria-hidden="true" tabindex="-1"></a>counts_st <span class="ot">&lt;-</span> <span class="fu">sapply</span>(counts<span class="sc">$</span>sample.info<span class="sc">$</span>SampleName, <span class="cf">function</span>(x) {</span>
<span id="cb25-3"><a href="analysis-protocol.html#cb25-3" aria-hidden="true" tabindex="-1"></a>  x.sample <span class="ot">&lt;-</span> sample.info[sample.info<span class="sc">$</span>Library.Name <span class="sc">==</span> x,]</span>
<span id="cb25-4"><a href="analysis-protocol.html#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="co">#get counts from cellCounts</span></span>
<span id="cb25-5"><a href="analysis-protocol.html#cb25-5" aria-hidden="true" tabindex="-1"></a>  x.counts <span class="ot">&lt;-</span> counts<span class="sc">$</span>counts[[x]]</span>
<span id="cb25-6"><a href="analysis-protocol.html#cb25-6" aria-hidden="true" tabindex="-1"></a>  <span class="co">#Change gene IDs to symbols</span></span>
<span id="cb25-7"><a href="analysis-protocol.html#cb25-7" aria-hidden="true" tabindex="-1"></a>  x.genes <span class="ot">&lt;-</span> ID2symbol.lst[<span class="fu">rownames</span>(x.counts)]</span>
<span id="cb25-8"><a href="analysis-protocol.html#cb25-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(x.counts)[<span class="fu">rownames</span>(x.counts) <span class="sc">%in%</span> <span class="fu">names</span>(ID2symbol.lst)] <span class="ot">&lt;-</span></span>
<span id="cb25-9"><a href="analysis-protocol.html#cb25-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unlist</span>(ID2symbol.lst[<span class="fu">rownames</span>(x.counts)], <span class="at">use.names =</span> F)</span>
<span id="cb25-10"><a href="analysis-protocol.html#cb25-10" aria-hidden="true" tabindex="-1"></a>  <span class="co">#create Seurat object</span></span>
<span id="cb25-11"><a href="analysis-protocol.html#cb25-11" aria-hidden="true" tabindex="-1"></a>  x.counts <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(x.counts,</span>
<span id="cb25-12"><a href="analysis-protocol.html#cb25-12" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">min.cells =</span> <span class="dv">0</span>)</span>
<span id="cb25-13"><a href="analysis-protocol.html#cb25-13" aria-hidden="true" tabindex="-1"></a>  <span class="co">#Calculate percentage of UMIs assigned to Mitochondrial genes in each Cell</span></span>
<span id="cb25-14"><a href="analysis-protocol.html#cb25-14" aria-hidden="true" tabindex="-1"></a>  x.counts[[<span class="st">&quot;percent.MT&quot;</span>]] <span class="ot">&lt;-</span></span>
<span id="cb25-15"><a href="analysis-protocol.html#cb25-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">PercentageFeatureSet</span>(x.counts, <span class="at">features =</span> genes.MT)</span>
<span id="cb25-16"><a href="analysis-protocol.html#cb25-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb25-17"><a href="analysis-protocol.html#cb25-17" aria-hidden="true" tabindex="-1"></a>  <span class="co">#Calculate percentage of UMIs assigned to Ribosomal genes in each Cell</span></span>
<span id="cb25-18"><a href="analysis-protocol.html#cb25-18" aria-hidden="true" tabindex="-1"></a>  x.counts[[<span class="st">&quot;percent.Ribo&quot;</span>]] <span class="ot">&lt;-</span></span>
<span id="cb25-19"><a href="analysis-protocol.html#cb25-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">PercentageFeatureSet</span>(x.counts, <span class="at">pattern =</span> <span class="st">&quot;^RP[SL]&quot;</span>)</span>
<span id="cb25-20"><a href="analysis-protocol.html#cb25-20" aria-hidden="true" tabindex="-1"></a>  <span class="co">#Add Sample metadata</span></span>
<span id="cb25-21"><a href="analysis-protocol.html#cb25-21" aria-hidden="true" tabindex="-1"></a>  x.counts[[<span class="st">&quot;cell.line&quot;</span>]] <span class="ot">&lt;-</span> x.sample<span class="sc">$</span>cell_line</span>
<span id="cb25-22"><a href="analysis-protocol.html#cb25-22" aria-hidden="true" tabindex="-1"></a>  x.counts[[<span class="st">&quot;tissue&quot;</span>]] <span class="ot">&lt;-</span> x.sample<span class="sc">$</span>tissue</span>
<span id="cb25-23"><a href="analysis-protocol.html#cb25-23" aria-hidden="true" tabindex="-1"></a>  x.counts[[<span class="st">&quot;Sample&quot;</span>]] <span class="ot">&lt;-</span> x.sample<span class="sc">$</span>Sample.Name</span>
<span id="cb25-24"><a href="analysis-protocol.html#cb25-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(x.counts)</span>
<span id="cb25-25"><a href="analysis-protocol.html#cb25-25" aria-hidden="true" tabindex="-1"></a>}, <span class="at">simplify =</span> F)</span>
<span id="cb25-26"><a href="analysis-protocol.html#cb25-26" aria-hidden="true" tabindex="-1"></a><span class="co">#Merge Seurat objects</span></span>
<span id="cb25-27"><a href="analysis-protocol.html#cb25-27" aria-hidden="true" tabindex="-1"></a>counts_st <span class="ot">&lt;-</span> <span class="fu">merge</span>(</span>
<span id="cb25-28"><a href="analysis-protocol.html#cb25-28" aria-hidden="true" tabindex="-1"></a>  counts_st<span class="sc">$</span>LLU_A_Seq1,</span>
<span id="cb25-29"><a href="analysis-protocol.html#cb25-29" aria-hidden="true" tabindex="-1"></a>  counts_st<span class="sc">$</span>LLU_B_Seq1,</span>
<span id="cb25-30"><a href="analysis-protocol.html#cb25-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">add.cell.ids =</span> counts<span class="sc">$</span>sample.info<span class="sc">$</span>SampleName,</span>
<span id="cb25-31"><a href="analysis-protocol.html#cb25-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">merge.data =</span> <span class="cn">TRUE</span></span>
<span id="cb25-32"><a href="analysis-protocol.html#cb25-32" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb25-33"><a href="analysis-protocol.html#cb25-33" aria-hidden="true" tabindex="-1"></a>counts_st</span></code></pre></div>
<pre><code>## An object of class Seurat 
## 28395 features across 4405 samples within 1 assay 
## Active assay: RNA (28395 features, 0 variable features)</code></pre>
</div>
<div id="quality-control-checks" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> Quality control checks</h2>
<div id="mitochondria-content" class="section level3" number="6.2.1">
<h3><span class="header-section-number">6.2.1</span> Mitochondria content</h3>
<p>We first look at the distribution of the Mitochondrial content to decide on the cut-off for filtering. The histogram below shows the distribution of the percentage of mitochondria content across all cells.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="analysis-protocol.html#cb27-1" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> <span class="fu">unique</span>(counts_st<span class="sc">@</span>meta.data<span class="sc">$</span>Sample)</span>
<span id="cb27-2"><a href="analysis-protocol.html#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(</span>
<span id="cb27-3"><a href="analysis-protocol.html#cb27-3" aria-hidden="true" tabindex="-1"></a>  counts_st<span class="sc">@</span>meta.data<span class="sc">$</span>percent.MT,</span>
<span id="cb27-4"><a href="analysis-protocol.html#cb27-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">&quot;Mitochondria content across all samples&quot;</span>,</span>
<span id="cb27-5"><a href="analysis-protocol.html#cb27-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">cex.main =</span> <span class="dv">1</span>,</span>
<span id="cb27-6"><a href="analysis-protocol.html#cb27-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">cex.lab =</span> <span class="fl">0.8</span>,</span>
<span id="cb27-7"><a href="analysis-protocol.html#cb27-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">xlab =</span> <span class="st">&quot;Percentage&quot;</span></span>
<span id="cb27-8"><a href="analysis-protocol.html#cb27-8" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<div class="figure"><span style="display:block;" id="fig:unnamed-chunk-18"></span>
<img src="scRNA-seq-tutorial_files/figure-html/unnamed-chunk-18-1.png" alt="Percentage of mitochondria" width="672" />
<p class="caption">
Figure 6.1: Percentage of mitochondria
</p>
</div>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="analysis-protocol.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>), <span class="at">oma =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">4</span>, <span class="dv">0</span>))</span>
<span id="cb28-2"><a href="analysis-protocol.html#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (s <span class="cf">in</span> samples)</span>
<span id="cb28-3"><a href="analysis-protocol.html#cb28-3" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb28-4"><a href="analysis-protocol.html#cb28-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hist</span>(</span>
<span id="cb28-5"><a href="analysis-protocol.html#cb28-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">subset</span>(counts_st<span class="sc">@</span>meta.data, <span class="at">subset =</span> Sample <span class="sc">==</span> s)<span class="sc">$</span>percent.MT,</span>
<span id="cb28-6"><a href="analysis-protocol.html#cb28-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">main =</span> <span class="fu">paste0</span>(s),</span>
<span id="cb28-7"><a href="analysis-protocol.html#cb28-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">&quot;Percentage&quot;</span>,</span>
<span id="cb28-8"><a href="analysis-protocol.html#cb28-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">cex.main =</span> <span class="dv">1</span>,</span>
<span id="cb28-9"><a href="analysis-protocol.html#cb28-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">cex.lab =</span> <span class="fl">0.9</span>,</span>
<span id="cb28-10"><a href="analysis-protocol.html#cb28-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">cex.axis =</span> <span class="fl">0.8</span></span>
<span id="cb28-11"><a href="analysis-protocol.html#cb28-11" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb28-12"><a href="analysis-protocol.html#cb28-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb28-13"><a href="analysis-protocol.html#cb28-13" aria-hidden="true" tabindex="-1"></a><span class="fu">title</span>(<span class="at">main =</span> <span class="st">&quot;Mitochondria content&quot;</span>,</span>
<span id="cb28-14"><a href="analysis-protocol.html#cb28-14" aria-hidden="true" tabindex="-1"></a>      <span class="at">cex.main =</span> <span class="dv">1</span>,</span>
<span id="cb28-15"><a href="analysis-protocol.html#cb28-15" aria-hidden="true" tabindex="-1"></a>      <span class="at">outer =</span> T)</span></code></pre></div>
<p><img src="scRNA-seq-tutorial_files/figure-html/unnamed-chunk-19-1.png" width="768" /></p>
</div>
<div id="ribosomal-content" class="section level3" number="6.2.2">
<h3><span class="header-section-number">6.2.2</span> Ribosomal content</h3>
<p>The histograms below show the distribution of the Ribosomal content across all samples.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="analysis-protocol.html#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(</span>
<span id="cb29-2"><a href="analysis-protocol.html#cb29-2" aria-hidden="true" tabindex="-1"></a>  counts_st<span class="sc">@</span>meta.data<span class="sc">$</span>percent.Ribo,</span>
<span id="cb29-3"><a href="analysis-protocol.html#cb29-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">xlab =</span> <span class="st">&quot;Percentage&quot;</span>,</span>
<span id="cb29-4"><a href="analysis-protocol.html#cb29-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">cex.main =</span> <span class="dv">1</span>,</span>
<span id="cb29-5"><a href="analysis-protocol.html#cb29-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">&quot;Ribosomal content across all samples&quot;</span>,</span>
<span id="cb29-6"><a href="analysis-protocol.html#cb29-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">cex.lab =</span> <span class="fl">0.8</span></span>
<span id="cb29-7"><a href="analysis-protocol.html#cb29-7" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="scRNA-seq-tutorial_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="analysis-protocol.html#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="at">oma =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">4</span>, <span class="dv">0</span>))</span>
<span id="cb30-2"><a href="analysis-protocol.html#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (s <span class="cf">in</span> samples)</span>
<span id="cb30-3"><a href="analysis-protocol.html#cb30-3" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb30-4"><a href="analysis-protocol.html#cb30-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hist</span>(</span>
<span id="cb30-5"><a href="analysis-protocol.html#cb30-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">subset</span>(counts_st<span class="sc">@</span>meta.data, <span class="at">subset =</span> Sample <span class="sc">==</span> s)<span class="sc">$</span>percent.Ribo,</span>
<span id="cb30-6"><a href="analysis-protocol.html#cb30-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">main =</span> s,</span>
<span id="cb30-7"><a href="analysis-protocol.html#cb30-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">&quot;Percentage&quot;</span>,</span>
<span id="cb30-8"><a href="analysis-protocol.html#cb30-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">cex.main =</span> <span class="dv">1</span>,</span>
<span id="cb30-9"><a href="analysis-protocol.html#cb30-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">cex.lab =</span> <span class="fl">0.9</span>,</span>
<span id="cb30-10"><a href="analysis-protocol.html#cb30-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">cex.axis =</span> <span class="fl">0.8</span></span>
<span id="cb30-11"><a href="analysis-protocol.html#cb30-11" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb30-12"><a href="analysis-protocol.html#cb30-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb30-13"><a href="analysis-protocol.html#cb30-13" aria-hidden="true" tabindex="-1"></a><span class="fu">title</span>(<span class="at">main =</span> <span class="st">&quot;Ribosomal content&quot;</span>,</span>
<span id="cb30-14"><a href="analysis-protocol.html#cb30-14" aria-hidden="true" tabindex="-1"></a>      <span class="at">cex.main =</span> <span class="dv">1</span>,</span>
<span id="cb30-15"><a href="analysis-protocol.html#cb30-15" aria-hidden="true" tabindex="-1"></a>      <span class="at">outer =</span> T)</span></code></pre></div>
<p><img src="scRNA-seq-tutorial_files/figure-html/unnamed-chunk-21-1.png" width="768" /></p>
</div>
<div id="number-of-detected-genes" class="section level3" number="6.2.3">
<h3><span class="header-section-number">6.2.3</span> Number of detected genes</h3>
<p>The histogram plot below shows the distribution of the number of detected genes across cells from all samples.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="analysis-protocol.html#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(</span>
<span id="cb31-2"><a href="analysis-protocol.html#cb31-2" aria-hidden="true" tabindex="-1"></a>  counts_st<span class="sc">@</span>meta.data<span class="sc">$</span>nFeature_RNA,</span>
<span id="cb31-3"><a href="analysis-protocol.html#cb31-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">&quot;Number of detected genes &quot;</span>,</span>
<span id="cb31-4"><a href="analysis-protocol.html#cb31-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">xlab =</span> <span class="st">&quot;Genes per cell&quot;</span>,</span>
<span id="cb31-5"><a href="analysis-protocol.html#cb31-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">cex.main =</span> <span class="dv">1</span>,</span>
<span id="cb31-6"><a href="analysis-protocol.html#cb31-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">cex.axis =</span> <span class="fl">0.9</span>,</span>
<span id="cb31-7"><a href="analysis-protocol.html#cb31-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">cex.lab =</span> <span class="fl">0.8</span></span>
<span id="cb31-8"><a href="analysis-protocol.html#cb31-8" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="scRNA-seq-tutorial_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="analysis-protocol.html#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="at">oma =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">4</span>, <span class="dv">0</span>))</span>
<span id="cb32-2"><a href="analysis-protocol.html#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (s <span class="cf">in</span> samples)</span>
<span id="cb32-3"><a href="analysis-protocol.html#cb32-3" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb32-4"><a href="analysis-protocol.html#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">hist</span>(</span>
<span id="cb32-5"><a href="analysis-protocol.html#cb32-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">subset</span>(counts_st<span class="sc">@</span>meta.data, <span class="at">subset =</span> Sample <span class="sc">==</span> s)<span class="sc">$</span>nFeature_RNA,</span>
<span id="cb32-6"><a href="analysis-protocol.html#cb32-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">main =</span> s,</span>
<span id="cb32-7"><a href="analysis-protocol.html#cb32-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">&quot;Detected genes&quot;</span>,</span>
<span id="cb32-8"><a href="analysis-protocol.html#cb32-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">cex.main =</span> <span class="dv">1</span>,</span>
<span id="cb32-9"><a href="analysis-protocol.html#cb32-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">cex.lab =</span> <span class="fl">0.9</span>,</span>
<span id="cb32-10"><a href="analysis-protocol.html#cb32-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">cex.axis =</span> <span class="fl">0.8</span></span>
<span id="cb32-11"><a href="analysis-protocol.html#cb32-11" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb32-12"><a href="analysis-protocol.html#cb32-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-13"><a href="analysis-protocol.html#cb32-13" aria-hidden="true" tabindex="-1"></a><span class="fu">title</span>(<span class="at">main =</span> <span class="st">&quot;Number of detected genes&quot;</span>,</span>
<span id="cb32-14"><a href="analysis-protocol.html#cb32-14" aria-hidden="true" tabindex="-1"></a>      <span class="at">outer =</span> T,</span>
<span id="cb32-15"><a href="analysis-protocol.html#cb32-15" aria-hidden="true" tabindex="-1"></a>      <span class="at">cex.main =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="scRNA-seq-tutorial_files/figure-html/unnamed-chunk-23-1.png" width="768" /></p>
</div>
</div>
<div id="filtering" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> Filtering</h2>
<div id="cell-filtering" class="section level3" number="6.3.1">
<h3><span class="header-section-number">6.3.1</span> Cell filtering</h3>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="analysis-protocol.html#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Filters used </span></span>
<span id="cb33-2"><a href="analysis-protocol.html#cb33-2" aria-hidden="true" tabindex="-1"></a>min.exp_genes<span class="ot">&lt;-</span><span class="dv">200</span> <span class="co">#minimum number of detected genes in each cell</span></span>
<span id="cb33-3"><a href="analysis-protocol.html#cb33-3" aria-hidden="true" tabindex="-1"></a>max.exp_genes<span class="ot">&lt;-</span><span class="dv">6000</span> <span class="co">#Maximum number of detected genes in each cell</span></span>
<span id="cb33-4"><a href="analysis-protocol.html#cb33-4" aria-hidden="true" tabindex="-1"></a>percent.mito <span class="ot">&lt;-</span> <span class="dv">20</span> <span class="co">#Percentage of mitochondrial content</span></span>
<span id="cb33-5"><a href="analysis-protocol.html#cb33-5" aria-hidden="true" tabindex="-1"></a>percent.ribos <span class="ot">&lt;-</span> <span class="dv">40</span> <span class="co"># Max percentage of Ribosomal content</span></span></code></pre></div>
<p>Based on the quality control checks above, the following filters are used to exclude cells that did not meet the criteria from the rest of the analysis</p>
<p>Number of cells before filtering for mitochondrial and ribosomal content: 4,405</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="analysis-protocol.html#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co">#exclude cells with a high content of mitochondrial and ribosomal content</span></span>
<span id="cb34-2"><a href="analysis-protocol.html#cb34-2" aria-hidden="true" tabindex="-1"></a>counts_st <span class="ot">&lt;-</span><span class="fu">subset</span>(counts_st, <span class="at">subset =</span> percent.MT <span class="sc">&lt;</span> percent.mito <span class="sc">&amp;</span></span>
<span id="cb34-3"><a href="analysis-protocol.html#cb34-3" aria-hidden="true" tabindex="-1"></a>           percent.Ribo <span class="sc">&lt;</span> percent.ribos)</span></code></pre></div>
<p>Number of cells after filtering for mitochondria and ribosomal content: 3,731
Next we exclude cells that had less than 200 and greater than 6,000</p>
<p>Number of cells remaining after filtering for cells with
low number of detected genes: 3,678
Number of cells left per sample</p>
<p>HCC1395 HCC1395BL
2794 884</p>
</div>
<div id="gene-filtering" class="section level3" number="6.3.2">
<h3><span class="header-section-number">6.3.2</span> Gene filtering</h3>
<p>Genes that failed to express (an expressed gene has at least 1 UMI count) in at least 3 cells in at least 1 sample together with Mitochondrial and Ribosomal genes are excluded from the rest of the analysis;</p>
<p>Number of genes before filter: 28,395</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="analysis-protocol.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Check genes expressed in at least 3 cells in at least one of the samples</span></span>
<span id="cb35-2"><a href="analysis-protocol.html#cb35-2" aria-hidden="true" tabindex="-1"></a>keep.genes <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">do.call</span>(cbind, <span class="fu">lapply</span>(samples, <span class="cf">function</span>(x)</span>
<span id="cb35-3"><a href="analysis-protocol.html#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowSums</span>(<span class="fu">as.matrix</span>(</span>
<span id="cb35-4"><a href="analysis-protocol.html#cb35-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">subset</span>(counts_st, Sample <span class="sc">==</span> x)<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>counts</span>
<span id="cb35-5"><a href="analysis-protocol.html#cb35-5" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">&gt;=</span> <span class="dv">3</span>)))<span class="sc">&gt;</span><span class="dv">0</span></span>
<span id="cb35-6"><a href="analysis-protocol.html#cb35-6" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(keep.genes)</span></code></pre></div>
<pre><code>## keep.genes
## FALSE  TRUE 
## 10516 17879</code></pre>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="analysis-protocol.html#cb37-1" aria-hidden="true" tabindex="-1"></a>counts_st <span class="ot">&lt;-</span> counts_st[keep.genes, ]</span></code></pre></div>
<pre><code>## 
## Number of genes after filtering for  genes that failed to express in at least 3 cells in at least 1 sample: 17,879</code></pre>
<p>Mitochondrial and Ribosomal genes are also excluded from the rest of the analysis.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="analysis-protocol.html#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co">#filter ribosomal and mitochondrial genes</span></span>
<span id="cb39-2"><a href="analysis-protocol.html#cb39-2" aria-hidden="true" tabindex="-1"></a>counts_st <span class="ot">&lt;-</span></span>
<span id="cb39-3"><a href="analysis-protocol.html#cb39-3" aria-hidden="true" tabindex="-1"></a>  counts_st[<span class="sc">!</span>(<span class="fu">rownames</span>(counts_st) <span class="sc">%in%</span> genes.MT) <span class="sc">&amp;</span></span>
<span id="cb39-4"><a href="analysis-protocol.html#cb39-4" aria-hidden="true" tabindex="-1"></a>       <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;^RP[SL]&quot;</span>, <span class="fu">rownames</span>(counts_st)),]</span></code></pre></div>
<p>Number of genes after filtering for mitochondrial and ribosomal genes: 17,712</p>
</div>
</div>
<div id="normalization" class="section level2" number="6.4">
<h2><span class="header-section-number">6.4</span> Normalization</h2>
<p>After removing unwanted cells from the dataset, the data was normalized using a global-scaling normalization method “LogNormalize” that normalizes the feature expression measurements for each cell by the total expression, multiplies this by a scale factor (10,000 by default), and log-transforms the result.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="analysis-protocol.html#cb40-1" aria-hidden="true" tabindex="-1"></a>counts_st <span class="ot">&lt;-</span><span class="fu">NormalizeData</span>(</span>
<span id="cb40-2"><a href="analysis-protocol.html#cb40-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">object =</span> counts_st,</span>
<span id="cb40-3"><a href="analysis-protocol.html#cb40-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">normalization.method =</span> <span class="st">&quot;LogNormalize&quot;</span>,</span>
<span id="cb40-4"><a href="analysis-protocol.html#cb40-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">scale.factor =</span> <span class="dv">10000</span></span>
<span id="cb40-5"><a href="analysis-protocol.html#cb40-5" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
</div>
<div id="dimension-reduction-and-clustering" class="section level2" number="6.5">
<h2><span class="header-section-number">6.5</span> Dimension reduction and clustering</h2>
<p>A subset of highly variable genes between cells were calculated using <em>FindVariableFeatures</em> function within Seurat.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="analysis-protocol.html#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Identification of highly variable features (feature selection)</span></span>
<span id="cb41-2"><a href="analysis-protocol.html#cb41-2" aria-hidden="true" tabindex="-1"></a>counts_st <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(<span class="at">object =</span> counts_st )</span></code></pre></div>
<p>The scatter plot below shows the top 10 identified variable genes and their average expression against variance</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="analysis-protocol.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify the 10 most highly variable genes</span></span>
<span id="cb42-2"><a href="analysis-protocol.html#cb42-2" aria-hidden="true" tabindex="-1"></a>top10 <span class="ot">&lt;-</span> <span class="fu">head</span>(<span class="fu">VariableFeatures</span>(counts_st), <span class="dv">10</span>)</span>
<span id="cb42-3"><a href="analysis-protocol.html#cb42-3" aria-hidden="true" tabindex="-1"></a><span class="co"># plot variable features with labels</span></span>
<span id="cb42-4"><a href="analysis-protocol.html#cb42-4" aria-hidden="true" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span> <span class="fu">VariableFeaturePlot</span>(counts_st)</span>
<span id="cb42-5"><a href="analysis-protocol.html#cb42-5" aria-hidden="true" tabindex="-1"></a>plot2 <span class="ot">&lt;-</span> <span class="fu">LabelPoints</span>(<span class="at">plot =</span> plot1, <span class="at">points =</span> top10, <span class="at">repel =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## When using repel, set xnudge and ynudge to 0 for optimal results</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="analysis-protocol.html#cb44-1" aria-hidden="true" tabindex="-1"></a>plot2</span></code></pre></div>
<p><img src="scRNA-seq-tutorial_files/figure-html/unnamed-chunk-37-1.png" width="768" /></p>
<p>A linear transformation (‘scaling’) was applied to the normalized data prior to dimensional reduction techniques using the <em>ScaleData</em> function in <strong>Seurat</strong>.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="analysis-protocol.html#cb45-1" aria-hidden="true" tabindex="-1"></a>counts_st <span class="ot">&lt;-</span>  <span class="fu">ScaleData</span>(</span>
<span id="cb45-2"><a href="analysis-protocol.html#cb45-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">object =</span> counts_st,</span>
<span id="cb45-3"><a href="analysis-protocol.html#cb45-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">features =</span> <span class="fu">rownames</span>(counts_st),</span>
<span id="cb45-4"><a href="analysis-protocol.html#cb45-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">verbose =</span> F)</span></code></pre></div>
<p>Principal component analysis was performed using the variable genes identified above.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="analysis-protocol.html#cb46-1" aria-hidden="true" tabindex="-1"></a>counts_st <span class="ot">&lt;-</span><span class="fu">RunPCA</span>(</span>
<span id="cb46-2"><a href="analysis-protocol.html#cb46-2" aria-hidden="true" tabindex="-1"></a>    counts_st,</span>
<span id="cb46-3"><a href="analysis-protocol.html#cb46-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">features =</span> <span class="fu">VariableFeatures</span>(<span class="at">object =</span> counts_st))</span></code></pre></div>
<pre><code>## PC_ 1 
## Positive:  S100A6, S100A10, TNFRSF12A, TSPO, TPM2, CAV1, TPM1, CALD1, ANXA2, RHEB 
##     ITGB1, LARP6, NQO1, HSBP1, SERPINE2, MGST1, FTH1P3, HSPB1, CYP1B1, KRT81 
##     DEF8, TSC22D1, LY6K, TNNT1, TGFBI, CAV2, CALU, DSTN, CCND1, TUBB3 
## Negative:  CD74, HLA-DRA, MS4A1, HLA-DPA1, LAPTM5, ARHGDIB, HLA-DPB1, IGHM, LSP1, LIMD2 
##     MIR155HG, CD70, CCND2, PDLIM1, CORO1A, BST2, CD79A, SNCA, CCL3, CD48 
##     RAC2, SRGN, TMSB4X, IGLC2, PLEK, CCR7, HCLS1, TNFRSF13B, CD53, IRF4 
## PC_ 2 
## Positive:  NME7, NDRG1, GADD45A, WIPI1, YPEL5, H2BC12, LURAP1L, GABARAPL1, GDF5, ZNF667-AS1 
##     ULK4, MYLK, KRT16, CCN1, FN1, LINC01315, NUPR1, BEX2, DNAJB9, PLK2 
##     SGK1, KRT75, CCN2, FBLIM1, MYC, MAP1LC3B, SELENOM, LINC01133, DUSP10, IL1RL1 
## Negative:  MKI67, CENPF, PLK1, HMGB2, CDC20, AURKA, CDCA3, UBE2C, CENPA, CCNA2 
##     GTSE1, HMMR, PTTG1, TPX2, NUSAP1, SMC4, TOP2A, CKS2, CENPE, CCNB1 
##     CDKN3, BIRC5, PRC1, CCNB2, ARL6IP1, DEPDC1, SGO2, ASPM, UBE2S, KPNA2 
## PC_ 3 
## Positive:  GADD45A, MARCKSL1, LAPTM4A, FDPS, ATP6V0B, PSMD8, PSMC2, HLA-C, PRMT1, TAGLN2 
##     CD44, TMBIM6, PAK1IP1, LGMN, DNAJA1, UBB, XRCC6, ENO1, TIMP1, RIPK2 
##     HACD3, DCAF13, UBC, TXNL1, HCCS, ADH5, CCT4, RTCB, DNAJB6, PGK1 
## Negative:  ULK4, ASPM, MKI67, C1orf56, CENPF, JCHAIN, TOP2A, GTSE1, NUSAP1, MXD3 
##     HMGB2, CDKN3, KIF14, CENPA, IGLL1, HMMR, IGLC7, BIRC5, PHKG1, IGLC5 
##     ITGA4, PRDM1, CENPE, CKAP2, TPX2, PRR11, DLGAP5, CDCA3, PIF1, IGLL5 
## PC_ 4 
## Positive:  ZFP36L1, DEK, BASP1, H4C3, STAG3, C12orf75, CCL22, SCG5, SERTAD4-AS1, CD82 
##     ATAD2, TNFSF4, CENPX, PRRX1, ARL4C, HCFC1R1, TNFAIP3, RAC3, FAM20C, DUSP4 
##     GTSE1, FTH1P3, ANKRD33B, SOX4, MARCKSL1, TNNT1, ANKRD11, PRR11, SMIM14, RUNX3 
## Negative:  UBB, PRDX1, PGK1, PSMA4, ENO1, PSMA3, EIF4A3, TMBIM6, ERP44, JCHAIN 
##     DCAF13, NPM1, CCT8, XRCC6, PSME2, IGLL1, IGLC7, IGLC5, MDH1, HSPA5 
##     HSPD1, HSP90AB1, CCT4, CCT5, HSP90B1, PSMC2, UBC, IGLL5, PRMT1, DNAJB11 
## PC_ 5 
## Positive:  C12orf75, JCHAIN, IGLC7, IGLL1, IGLC5, IGLL5, FN1, FTL, ITGA4, SPHK1 
##     PRDM1, TNFRSF17, GLCCI1, SSR3, IGLL3P, RASSF6, GUSBP11, LINC01480, CLEC2B, ST6GAL1 
##     GLIPR1, ISG15, XBP1, POU2AF1, FAM107B, SDF2L1, RRAGD, CCDC69, MEF2C, CYTOR 
## Negative:  MMP7, LTA, AKR1C2, WDR91, BCL2A1, CCL22, AKR1C1, PRDX1, ENO1, SSTR2 
##     HSP90AB1, LDHA, AKR1C3, TRAF1, TNFRSF8, STAG3, PKM, HSPD1, EBI3, AKR1B10 
##     MUC12-AS1, LINC00158, ICAM1, HSP90AA1, ZBTB32, RHOV, AKR1C4, NFKBIA, CCT5, BCAT1</code></pre>
<p>We use the <em>JackStraw</em> function in <strong>Seurat</strong> to establish the dimensionality of the dataset. This is done by randomly selecting a subset of genes and their significance in a computed PC.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="analysis-protocol.html#cb48-1" aria-hidden="true" tabindex="-1"></a>counts_st <span class="ot">&lt;-</span> <span class="fu">JackStraw</span>(counts_st, <span class="at">verbose =</span> F, <span class="at">dims =</span> <span class="dv">20</span>)</span>
<span id="cb48-2"><a href="analysis-protocol.html#cb48-2" aria-hidden="true" tabindex="-1"></a>counts_st <span class="ot">&lt;-</span></span>
<span id="cb48-3"><a href="analysis-protocol.html#cb48-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ScoreJackStraw</span>(counts_st, <span class="at">reduction =</span> <span class="st">&quot;pca&quot;</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>)</span></code></pre></div>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="analysis-protocol.html#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">JackStrawPlot</span>(counts_st, <span class="at">reduction =</span> <span class="st">&quot;pca&quot;</span>, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>)</span></code></pre></div>
<pre><code>## Warning: Removed 28000 rows containing missing values (geom_point).</code></pre>
<p><img src="scRNA-seq-tutorial_files/figure-html/elbowplot1-1.png" width="672" /></p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="analysis-protocol.html#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ElbowPlot</span>(counts_st)</span></code></pre></div>
<p><img src="scRNA-seq-tutorial_files/figure-html/elbowplot1-2.png" width="672" /></p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="analysis-protocol.html#cb52-1" aria-hidden="true" tabindex="-1"></a>res<span class="ot">&lt;-</span><span class="fl">0.5</span></span>
<span id="cb52-2"><a href="analysis-protocol.html#cb52-2" aria-hidden="true" tabindex="-1"></a>sel.dims<span class="ot">&lt;-</span><span class="dv">10</span></span></code></pre></div>
<p>10 dimensions are used to establish the number of clusters with a resolution of 0.5</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="analysis-protocol.html#cb53-1" aria-hidden="true" tabindex="-1"></a>counts_st <span class="ot">&lt;-</span><span class="fu">FindNeighbors</span>(counts_st,</span>
<span id="cb53-2"><a href="analysis-protocol.html#cb53-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">reduction =</span> <span class="st">&quot;pca&quot;</span>,</span>
<span id="cb53-3"><a href="analysis-protocol.html#cb53-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">verbose =</span> F,</span>
<span id="cb53-4"><a href="analysis-protocol.html#cb53-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span>sel.dims)</span></code></pre></div>
<p>Next we run the <strong>FindClusters</strong> to identify clusters of cells using a resolution of 0.5.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="analysis-protocol.html#cb54-1" aria-hidden="true" tabindex="-1"></a>counts_st <span class="ot">&lt;-</span></span>
<span id="cb54-2"><a href="analysis-protocol.html#cb54-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">FindClusters</span>(</span>
<span id="cb54-3"><a href="analysis-protocol.html#cb54-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">object =</span> counts_st,</span>
<span id="cb54-4"><a href="analysis-protocol.html#cb54-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">resolution =</span> res,</span>
<span id="cb54-5"><a href="analysis-protocol.html#cb54-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">verbose =</span> <span class="dv">0</span></span>
<span id="cb54-6"><a href="analysis-protocol.html#cb54-6" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
<p>t-SNE dimensionality reduction on is run on the identified variable features.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="analysis-protocol.html#cb55-1" aria-hidden="true" tabindex="-1"></a>counts_st <span class="ot">&lt;-</span> <span class="fu">RunTSNE</span>(counts_st, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span>sel.dims)</span></code></pre></div>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="analysis-protocol.html#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(</span>
<span id="cb56-2"><a href="analysis-protocol.html#cb56-2" aria-hidden="true" tabindex="-1"></a>  counts_st,</span>
<span id="cb56-3"><a href="analysis-protocol.html#cb56-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">reduction =</span> <span class="st">&quot;tsne&quot;</span>,</span>
<span id="cb56-4"><a href="analysis-protocol.html#cb56-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">label =</span> T,</span>
<span id="cb56-5"><a href="analysis-protocol.html#cb56-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">size =</span> <span class="fl">0.5</span>,</span>
<span id="cb56-6"><a href="analysis-protocol.html#cb56-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">repel =</span> T</span>
<span id="cb56-7"><a href="analysis-protocol.html#cb56-7" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="scRNA-seq-tutorial_files/figure-html/unnamed-chunk-42-1.png" width="960" /></p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="analysis-protocol.html#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(</span>
<span id="cb57-2"><a href="analysis-protocol.html#cb57-2" aria-hidden="true" tabindex="-1"></a>  counts_st,</span>
<span id="cb57-3"><a href="analysis-protocol.html#cb57-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">reduction =</span> <span class="st">&quot;tsne&quot;</span>,</span>
<span id="cb57-4"><a href="analysis-protocol.html#cb57-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">label =</span> T,</span>
<span id="cb57-5"><a href="analysis-protocol.html#cb57-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">group.by =</span> <span class="st">&quot;Sample&quot;</span>,</span>
<span id="cb57-6"><a href="analysis-protocol.html#cb57-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">size =</span> <span class="fl">0.5</span>,</span>
<span id="cb57-7"><a href="analysis-protocol.html#cb57-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">repel =</span> T</span>
<span id="cb57-8"><a href="analysis-protocol.html#cb57-8" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="scRNA-seq-tutorial_files/figure-html/unnamed-chunk-43-1.png" width="960" /></p>
<p>Optionally, we can also perform dimension reductionality reduction using UMAP</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="analysis-protocol.html#cb58-1" aria-hidden="true" tabindex="-1"></a>counts_st <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(counts_st, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span>sel.dims,<span class="at">verbose =</span> F)</span></code></pre></div>
<pre><code>## Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric
## To use Python UMAP via reticulate, set umap.method to &#39;umap-learn&#39; and metric to &#39;correlation&#39;
## This message will be shown once per session</code></pre>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="analysis-protocol.html#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(</span>
<span id="cb60-2"><a href="analysis-protocol.html#cb60-2" aria-hidden="true" tabindex="-1"></a>  counts_st,</span>
<span id="cb60-3"><a href="analysis-protocol.html#cb60-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>,</span>
<span id="cb60-4"><a href="analysis-protocol.html#cb60-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">label =</span> T,</span>
<span id="cb60-5"><a href="analysis-protocol.html#cb60-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">size =</span> <span class="fl">0.5</span>,</span>
<span id="cb60-6"><a href="analysis-protocol.html#cb60-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">repel =</span> T</span>
<span id="cb60-7"><a href="analysis-protocol.html#cb60-7" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="scRNA-seq-tutorial_files/figure-html/unnamed-chunk-44-1.png" width="768" /></p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="analysis-protocol.html#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(</span>
<span id="cb61-2"><a href="analysis-protocol.html#cb61-2" aria-hidden="true" tabindex="-1"></a>  counts_st,</span>
<span id="cb61-3"><a href="analysis-protocol.html#cb61-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>,</span>
<span id="cb61-4"><a href="analysis-protocol.html#cb61-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">label =</span> T,</span>
<span id="cb61-5"><a href="analysis-protocol.html#cb61-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">group.by =</span> <span class="st">&quot;Sample&quot;</span>,</span>
<span id="cb61-6"><a href="analysis-protocol.html#cb61-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">size =</span> <span class="fl">0.5</span>,</span>
<span id="cb61-7"><a href="analysis-protocol.html#cb61-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">repel =</span> T</span>
<span id="cb61-8"><a href="analysis-protocol.html#cb61-8" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="scRNA-seq-tutorial_files/figure-html/unnamed-chunk-45-1.png" width="768" />
## Differential expression analysis between clusters</p>
<p>To identify differentially expressed genes between clusters we can use the <em>FindMarker</em> function in Seurat. As an example, DE genes between clusters 0 and 1 are computed below.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="analysis-protocol.html#cb62-1" aria-hidden="true" tabindex="-1"></a>DE.cluster0_1 <span class="ot">&lt;-</span> <span class="fu">FindMarkers</span>(counts_st,</span>
<span id="cb62-2"><a href="analysis-protocol.html#cb62-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">ident.1 =</span> <span class="dv">0</span>,</span>
<span id="cb62-3"><a href="analysis-protocol.html#cb62-3" aria-hidden="true" tabindex="-1"></a>                             <span class="at">ident.2 =</span> <span class="dv">1</span>,</span>
<span id="cb62-4"><a href="analysis-protocol.html#cb62-4" aria-hidden="true" tabindex="-1"></a>                             <span class="at">verbose =</span> F)</span>
<span id="cb62-5"><a href="analysis-protocol.html#cb62-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(DE.cluster0_1[<span class="fu">order</span>(<span class="fu">abs</span>(DE.cluster0_1<span class="sc">$</span>avg_log2FC), <span class="at">decreasing =</span> T), ])</span></code></pre></div>
<pre><code>##                p_val avg_log2FC pct.1 pct.2     p_val_adj
## CD74    0.000000e+00  -5.891077 0.008 1.000  0.000000e+00
## IGHM    0.000000e+00  -5.729445 0.000 0.998  0.000000e+00
## S100A6 3.200845e-306   5.592588 1.000 0.047 5.669336e-302
## TMSB4X 2.191709e-273  -4.934792 0.969 1.000 3.881954e-269
## CCL3   5.926187e-299  -4.861986 0.000 0.945 1.049646e-294
## KRT81  1.055291e-293   4.609987 0.964 0.000 1.869131e-289</code></pre>
<p>Marker genes which can be used to uniquely identify each of the clusters are identified using the <em>FindAllMarkers</em> function.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="analysis-protocol.html#cb64-1" aria-hidden="true" tabindex="-1"></a>all.markers<span class="ot">&lt;-</span><span class="fu">FindAllMarkers</span>(counts_st,<span class="at">verbose =</span> F)</span></code></pre></div>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="analysis-protocol.html#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb65-2"><a href="analysis-protocol.html#cb65-2" aria-hidden="true" tabindex="-1"></a>all.markers <span class="sc">%&gt;%</span></span>
<span id="cb65-3"><a href="analysis-protocol.html#cb65-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span></span>
<span id="cb65-4"><a href="analysis-protocol.html#cb65-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_max</span>(<span class="at">n =</span> <span class="dv">5</span>, <span class="at">order_by =</span> avg_log2FC)</span></code></pre></div>
<pre><code>## # A tibble: 35 × 7
## # Groups:   cluster [7]
##        p_val avg_log2FC pct.1 pct.2 p_val_adj cluster gene   
##        &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;fct&gt;   &lt;chr&gt;  
##  1 1.61e-155      1.23  0.94  0.525 2.86e-151 0       IGFBP3 
##  2 4.07e- 65      1.17  0.417 0.149 7.21e- 61 0       IGFBP5 
##  3 2.12e-147      1.06  0.998 0.714 3.75e-143 0       TPM1   
##  4 2.31e-202      0.938 0.994 0.869 4.09e-198 0       CSNK2B 
##  5 1.78e-131      0.901 0.998 0.665 3.15e-127 0       TPM2   
##  6 0              4.79  0.998 0.021 0         1       IGHM   
##  7 0              4.63  1     0.037 0         1       CD74   
##  8 0              4.22  0.945 0.015 0         1       CCL3   
##  9 0              4.08  1     0.959 0         1       TMSB4X 
## 10 0              3.48  0.999 0.033 0         1       HLA-DRA
## # … with 25 more rows</code></pre>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="analysis-protocol.html#cb67-1" aria-hidden="true" tabindex="-1"></a>top10.markers<span class="ot">&lt;-</span>all.markers <span class="sc">%&gt;%</span></span>
<span id="cb67-2"><a href="analysis-protocol.html#cb67-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span></span>
<span id="cb67-3"><a href="analysis-protocol.html#cb67-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">slice_max</span>(<span class="at">n =</span> <span class="dv">10</span>, <span class="at">order_by =</span> avg_log2FC)</span></code></pre></div>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="analysis-protocol.html#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DoHeatmap</span>(counts_st, <span class="at">features =</span> top10.markers<span class="sc">$</span>gene)</span></code></pre></div>
<p><img src="scRNA-seq-tutorial_files/figure-html/unnamed-chunk-51-1.png" width="960" /></p>
</div>
<div id="cell-annotation" class="section level2" number="6.6">
<h2><span class="header-section-number">6.6</span> Cell annotation</h2>
<p>An unbiased cell type recognition is performed using <strong>SingleR</strong>. <strong>celldex</strong> has a range of annotations derived from Bulk RNA-seq data that can be used to annotate the identified clusters above. Here, we use the Human Primary Cell Atlas database as an example.</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="analysis-protocol.html#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SingleR)</span></code></pre></div>
<pre><code>## Warning: package &#39;MatrixGenerics&#39; was built under R version 4.1.1</code></pre>
<pre><code>## Warning: package &#39;GenomeInfoDb&#39; was built under R version 4.1.1</code></pre>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="analysis-protocol.html#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SingleCellExperiment)</span>
<span id="cb72-2"><a href="analysis-protocol.html#cb72-2" aria-hidden="true" tabindex="-1"></a>cell.sce<span class="ot">&lt;-</span> <span class="fu">as.SingleCellExperiment</span>(counts_st)</span>
<span id="cb72-3"><a href="analysis-protocol.html#cb72-3" aria-hidden="true" tabindex="-1"></a>annot<span class="ot">&lt;-</span>celldex<span class="sc">::</span><span class="fu">HumanPrimaryCellAtlasData</span>()</span>
<span id="cb72-4"><a href="analysis-protocol.html#cb72-4" aria-hidden="true" tabindex="-1"></a>cell.annots <span class="ot">&lt;-</span> <span class="fu">SingleR</span>(</span>
<span id="cb72-5"><a href="analysis-protocol.html#cb72-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">test =</span> cell.sce,</span>
<span id="cb72-6"><a href="analysis-protocol.html#cb72-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">ref =</span> annot,</span>
<span id="cb72-7"><a href="analysis-protocol.html#cb72-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels =</span> annot<span class="sc">$</span>label.main)</span>
<span id="cb72-8"><a href="analysis-protocol.html#cb72-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-9"><a href="analysis-protocol.html#cb72-9" aria-hidden="true" tabindex="-1"></a>cell.annots.fine<span class="ot">&lt;-</span><span class="fu">SingleR</span>(</span>
<span id="cb72-10"><a href="analysis-protocol.html#cb72-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">test =</span> cell.sce,</span>
<span id="cb72-11"><a href="analysis-protocol.html#cb72-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">ref =</span> annot,</span>
<span id="cb72-12"><a href="analysis-protocol.html#cb72-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">labels =</span> annot<span class="sc">$</span>label.fine)</span></code></pre></div>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="analysis-protocol.html#cb73-1" aria-hidden="true" tabindex="-1"></a>counts_st <span class="ot">&lt;-</span></span>
<span id="cb73-2"><a href="analysis-protocol.html#cb73-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">AddMetaData</span>(counts_st, cell.annots[<span class="fu">rownames</span>(counts_st<span class="sc">@</span>meta.data), <span class="st">&quot;labels&quot;</span>], <span class="st">&quot;Annot.main&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="analysis-protocol.html#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(</span>
<span id="cb74-2"><a href="analysis-protocol.html#cb74-2" aria-hidden="true" tabindex="-1"></a>    counts_st,</span>
<span id="cb74-3"><a href="analysis-protocol.html#cb74-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">reduction =</span> <span class="st">&quot;tsne&quot;</span>,</span>
<span id="cb74-4"><a href="analysis-protocol.html#cb74-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">group.by =</span> <span class="st">&quot;Annot.main&quot;</span>,</span>
<span id="cb74-5"><a href="analysis-protocol.html#cb74-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> T,</span>
<span id="cb74-6"><a href="analysis-protocol.html#cb74-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">repel =</span> T,</span>
<span id="cb74-7"><a href="analysis-protocol.html#cb74-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">pt.size =</span> <span class="fl">0.1</span>,</span>
<span id="cb74-8"><a href="analysis-protocol.html#cb74-8" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
<pre><code>## Warning: ggrepel: 5 unlabeled data points (too many overlaps). Consider
## increasing max.overlaps</code></pre>
<p><img src="scRNA-seq-tutorial_files/figure-html/unnamed-chunk-56-1.png" width="960" /></p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="analysis-protocol.html#cb76-1" aria-hidden="true" tabindex="-1"></a>counts_st <span class="ot">&lt;-</span><span class="fu">AddMetaData</span>(counts_st, cell.annots[</span>
<span id="cb76-2"><a href="analysis-protocol.html#cb76-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(counts_st<span class="sc">@</span>meta.data), <span class="st">&quot;labels&quot;</span>], <span class="st">&quot;Annot.fine&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="analysis-protocol.html#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(</span>
<span id="cb77-2"><a href="analysis-protocol.html#cb77-2" aria-hidden="true" tabindex="-1"></a>    counts_st,</span>
<span id="cb77-3"><a href="analysis-protocol.html#cb77-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">reduction =</span> <span class="st">&quot;tsne&quot;</span>,</span>
<span id="cb77-4"><a href="analysis-protocol.html#cb77-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">group.by =</span> <span class="st">&quot;Annot.fine&quot;</span>,</span>
<span id="cb77-5"><a href="analysis-protocol.html#cb77-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> T,</span>
<span id="cb77-6"><a href="analysis-protocol.html#cb77-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">repel =</span> T,</span>
<span id="cb77-7"><a href="analysis-protocol.html#cb77-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">pt.size =</span> <span class="fl">0.1</span>)</span></code></pre></div>
<pre><code>## Warning: ggrepel: 5 unlabeled data points (too many overlaps). Consider
## increasing max.overlaps</code></pre>
<p><img src="scRNA-seq-tutorial_files/figure-html/unnamed-chunk-58-1.png" width="960" /></p>
</div>
<div id="trajectory-analysis" class="section level2" number="6.7">
<h2><span class="header-section-number">6.7</span> Trajectory analysis</h2>
<p>Pseudotime analysis of the tuft cells identified in the dataset was performed using <strong>Monocle2</strong>.</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="analysis-protocol.html#cb79-1" aria-hidden="true" tabindex="-1"></a>mono2.learn.traject <span class="ot">&lt;-</span></span>
<span id="cb79-2"><a href="analysis-protocol.html#cb79-2" aria-hidden="true" tabindex="-1"></a>  <span class="cf">function</span>(X_counts,</span>
<span id="cb79-3"><a href="analysis-protocol.html#cb79-3" aria-hidden="true" tabindex="-1"></a>           these.cell.types) {</span>
<span id="cb79-4"><a href="analysis-protocol.html#cb79-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">library</span>(monocle) <span class="co"># It has to be Monocle 2.</span></span>
<span id="cb79-5"><a href="analysis-protocol.html#cb79-5" aria-hidden="true" tabindex="-1"></a>    rds.fname <span class="ot">&lt;-</span><span class="st">&quot;Trajectory-cds.rds&quot;</span></span>
<span id="cb79-6"><a href="analysis-protocol.html#cb79-6" aria-hidden="true" tabindex="-1"></a>    gsndf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene_short_name =</span> <span class="fu">rownames</span>(X_counts))</span>
<span id="cb79-7"><a href="analysis-protocol.html#cb79-7" aria-hidden="true" tabindex="-1"></a>    csndf <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">cell.type =</span> these.cell.types)</span>
<span id="cb79-8"><a href="analysis-protocol.html#cb79-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(gsndf) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(X_counts)</span>
<span id="cb79-9"><a href="analysis-protocol.html#cb79-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rownames</span>(csndf) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(X_counts)</span>
<span id="cb79-10"><a href="analysis-protocol.html#cb79-10" aria-hidden="true" tabindex="-1"></a>    pd <span class="ot">&lt;-</span> <span class="fu">new</span>(<span class="st">&quot;AnnotatedDataFrame&quot;</span>, <span class="at">data =</span> csndf)</span>
<span id="cb79-11"><a href="analysis-protocol.html#cb79-11" aria-hidden="true" tabindex="-1"></a>    fd <span class="ot">&lt;-</span> <span class="fu">new</span>(<span class="st">&quot;AnnotatedDataFrame&quot;</span>, <span class="at">data =</span> gsndf)</span>
<span id="cb79-12"><a href="analysis-protocol.html#cb79-12" aria-hidden="true" tabindex="-1"></a>    cds <span class="ot">&lt;-</span></span>
<span id="cb79-13"><a href="analysis-protocol.html#cb79-13" aria-hidden="true" tabindex="-1"></a>      <span class="fu">newCellDataSet</span>(</span>
<span id="cb79-14"><a href="analysis-protocol.html#cb79-14" aria-hidden="true" tabindex="-1"></a>        X_counts,</span>
<span id="cb79-15"><a href="analysis-protocol.html#cb79-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">phenoData =</span> pd,</span>
<span id="cb79-16"><a href="analysis-protocol.html#cb79-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">featureData =</span> fd,</span>
<span id="cb79-17"><a href="analysis-protocol.html#cb79-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">expressionFamily =</span> <span class="fu">negbinomial.size</span>()</span>
<span id="cb79-18"><a href="analysis-protocol.html#cb79-18" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb79-19"><a href="analysis-protocol.html#cb79-19" aria-hidden="true" tabindex="-1"></a>    cds <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(cds)</span>
<span id="cb79-20"><a href="analysis-protocol.html#cb79-20" aria-hidden="true" tabindex="-1"></a>    cds <span class="ot">&lt;-</span> <span class="fu">estimateDispersions</span>(cds)</span>
<span id="cb79-21"><a href="analysis-protocol.html#cb79-21" aria-hidden="true" tabindex="-1"></a>    cds <span class="ot">&lt;-</span> <span class="fu">detectGenes</span>(cds, <span class="at">min_expr =</span> <span class="fl">0.1</span>)</span>
<span id="cb79-22"><a href="analysis-protocol.html#cb79-22" aria-hidden="true" tabindex="-1"></a>    disp_table <span class="ot">&lt;-</span> <span class="fu">dispersionTable</span>(cds)</span>
<span id="cb79-23"><a href="analysis-protocol.html#cb79-23" aria-hidden="true" tabindex="-1"></a>    ordering_genes <span class="ot">&lt;-</span> <span class="fu">subset</span>(disp_table, mean_expression <span class="sc">&gt;=</span> <span class="fl">0.1</span>)</span>
<span id="cb79-24"><a href="analysis-protocol.html#cb79-24" aria-hidden="true" tabindex="-1"></a>    cds <span class="ot">&lt;-</span> <span class="fu">setOrderingFilter</span>(cds, ordering_genes)</span>
<span id="cb79-25"><a href="analysis-protocol.html#cb79-25" aria-hidden="true" tabindex="-1"></a>    cds <span class="ot">&lt;-</span> <span class="fu">reduceDimension</span>(cds)</span>
<span id="cb79-26"><a href="analysis-protocol.html#cb79-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">saveRDS</span>(cds , rds.fname)</span>
<span id="cb79-27"><a href="analysis-protocol.html#cb79-27" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="analysis-protocol.html#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mono2.learn.traject</span>(</span>
<span id="cb80-2"><a href="analysis-protocol.html#cb80-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">X_counts =</span> <span class="fu">as.matrix</span>(counts_st<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">@</span>counts),</span>
<span id="cb80-3"><a href="analysis-protocol.html#cb80-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">these.cell.types =</span> counts_st[[]]<span class="sc">$</span>seurat_clusters)</span></code></pre></div>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="analysis-protocol.html#cb81-1" aria-hidden="true" tabindex="-1"></a>mono.rds <span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="st">&quot;Trajectory-cds.rds&quot;</span>)</span>
<span id="cb81-2"><a href="analysis-protocol.html#cb81-2" aria-hidden="true" tabindex="-1"></a>mono.Tree <span class="ot">&lt;-</span> <span class="fu">t</span>(mono.rds<span class="sc">@</span>reducedDimS)</span></code></pre></div>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="analysis-protocol.html#cb82-1" aria-hidden="true" tabindex="-1"></a>do.Traj.lines <span class="ot">&lt;-</span> <span class="cf">function</span>(traj) {</span>
<span id="cb82-2"><a href="analysis-protocol.html#cb82-2" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(traj<span class="sc">@</span>minSpanningTree[[]])) {</span>
<span id="cb82-3"><a href="analysis-protocol.html#cb82-3" aria-hidden="true" tabindex="-1"></a>    p1no <span class="ot">&lt;-</span> <span class="fu">unlist</span>(traj<span class="sc">@</span>minSpanningTree[[i]])[<span class="dv">1</span>]</span>
<span id="cb82-4"><a href="analysis-protocol.html#cb82-4" aria-hidden="true" tabindex="-1"></a>    p2no <span class="ot">&lt;-</span> <span class="fu">unlist</span>(traj<span class="sc">@</span>minSpanningTree[[i]])[<span class="dv">2</span>]</span>
<span id="cb82-5"><a href="analysis-protocol.html#cb82-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lines</span>(</span>
<span id="cb82-6"><a href="analysis-protocol.html#cb82-6" aria-hidden="true" tabindex="-1"></a>      <span class="fu">c</span>(traj<span class="sc">@</span>reducedDimK[<span class="dv">1</span>, p1no],</span>
<span id="cb82-7"><a href="analysis-protocol.html#cb82-7" aria-hidden="true" tabindex="-1"></a>        traj<span class="sc">@</span>reducedDimK[<span class="dv">1</span>, p2no]),</span>
<span id="cb82-8"><a href="analysis-protocol.html#cb82-8" aria-hidden="true" tabindex="-1"></a>      <span class="fu">c</span>(traj<span class="sc">@</span>reducedDimK[<span class="dv">2</span>, p1no],</span>
<span id="cb82-9"><a href="analysis-protocol.html#cb82-9" aria-hidden="true" tabindex="-1"></a>        traj<span class="sc">@</span>reducedDimK[<span class="dv">2</span>, p2no]),</span>
<span id="cb82-10"><a href="analysis-protocol.html#cb82-10" aria-hidden="true" tabindex="-1"></a>      <span class="at">lwd =</span> <span class="dv">2</span></span>
<span id="cb82-11"><a href="analysis-protocol.html#cb82-11" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb82-12"><a href="analysis-protocol.html#cb82-12" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb82-13"><a href="analysis-protocol.html#cb82-13" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="analysis-protocol.html#cb83-1" aria-hidden="true" tabindex="-1"></a>clusters <span class="ot">&lt;-</span> counts_st[[]]<span class="sc">$</span>seurat_clusters</span>
<span id="cb83-2"><a href="analysis-protocol.html#cb83-2" aria-hidden="true" tabindex="-1"></a>cols.traj <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;green&quot;</span>, <span class="st">&quot;blue&quot;</span>,<span class="st">&quot;orange&quot;</span>,<span class="st">&quot;purple&quot;</span>,<span class="st">&quot;pink&quot;</span>)</span>
<span id="cb83-3"><a href="analysis-protocol.html#cb83-3" aria-hidden="true" tabindex="-1"></a>traj.cols <span class="ot">&lt;-</span>  cols.traj[clusters]</span>
<span id="cb83-4"><a href="analysis-protocol.html#cb83-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(</span>
<span id="cb83-5"><a href="analysis-protocol.html#cb83-5" aria-hidden="true" tabindex="-1"></a>    mono.Tree,</span>
<span id="cb83-6"><a href="analysis-protocol.html#cb83-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">cex =</span> <span class="fl">0.7</span>,</span>
<span id="cb83-7"><a href="analysis-protocol.html#cb83-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">pch =</span> <span class="dv">16</span>,</span>
<span id="cb83-8"><a href="analysis-protocol.html#cb83-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">bg =</span> traj.cols,</span>
<span id="cb83-9"><a href="analysis-protocol.html#cb83-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">col =</span> traj.cols,</span>
<span id="cb83-10"><a href="analysis-protocol.html#cb83-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">&quot;Component 1&quot;</span>,</span>
<span id="cb83-11"><a href="analysis-protocol.html#cb83-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylab =</span> <span class="st">&quot;Component 2&quot;</span>,</span>
<span id="cb83-12"><a href="analysis-protocol.html#cb83-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">cex.axis =</span> <span class="fl">0.8</span>,</span>
<span id="cb83-13"><a href="analysis-protocol.html#cb83-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">cex.lab =</span> <span class="fl">0.9</span></span>
<span id="cb83-14"><a href="analysis-protocol.html#cb83-14" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb83-15"><a href="analysis-protocol.html#cb83-15" aria-hidden="true" tabindex="-1"></a><span class="fu">do.Traj.lines</span>(mono.rds)</span>
<span id="cb83-16"><a href="analysis-protocol.html#cb83-16" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(</span>
<span id="cb83-17"><a href="analysis-protocol.html#cb83-17" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;bottomright&quot;</span>,</span>
<span id="cb83-18"><a href="analysis-protocol.html#cb83-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">legend =</span> <span class="fu">levels</span>(clusters),</span>
<span id="cb83-19"><a href="analysis-protocol.html#cb83-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">pch =</span> <span class="dv">16</span>,</span>
<span id="cb83-20"><a href="analysis-protocol.html#cb83-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">col =</span> cols.traj,</span>
<span id="cb83-21"><a href="analysis-protocol.html#cb83-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">bty =</span> <span class="st">&quot;n&quot;</span></span>
<span id="cb83-22"><a href="analysis-protocol.html#cb83-22" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="scRNA-seq-tutorial_files/figure-html/unnamed-chunk-63-1.png" width="768" /></p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="analysis-protocol.html#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.1.0 (2021-05-18)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Big Sur 10.16
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] SingleR_1.6.1               SummarizedExperiment_1.22.0
##  [3] GenomicRanges_1.44.0        GenomeInfoDb_1.28.4        
##  [5] MatrixGenerics_1.4.3        matrixStats_0.60.1         
##  [7] dplyr_1.0.7                 org.Hs.eg.db_3.13.0        
##  [9] AnnotationDbi_1.54.1        IRanges_2.26.0             
## [11] S4Vectors_0.30.0            Biobase_2.52.0             
## [13] BiocGenerics_0.38.0         SeuratObject_4.0.2         
## [15] Seurat_4.0.4                Rsubread_2.6.4             
## 
## loaded via a namespace (and not attached):
##   [1] VGAM_1.1-5                plyr_1.8.6               
##   [3] igraph_1.2.6              lazyeval_0.2.2           
##   [5] splines_4.1.0             densityClust_0.3         
##   [7] BiocParallel_1.26.2       listenv_0.8.0            
##   [9] scattermore_0.7           fastICA_1.2-2            
##  [11] ggplot2_3.3.5             digest_0.6.27            
##  [13] htmltools_0.5.2           viridis_0.6.1            
##  [15] fansi_0.5.0               magrittr_2.0.1           
##  [17] memoise_2.0.0             ScaledMatrix_1.0.0       
##  [19] tensor_1.5                cluster_2.1.2            
##  [21] ROCR_1.0-11               limma_3.48.3             
##  [23] globals_0.14.0            Biostrings_2.60.2        
##  [25] docopt_0.7.1              spatstat.sparse_2.0-0    
##  [27] colorspace_2.0-2          blob_1.2.2               
##  [29] ggrepel_0.9.1             xfun_0.25                
##  [31] sparsesvd_0.2             crayon_1.4.1             
##  [33] RCurl_1.98-1.4            jsonlite_1.7.2           
##  [35] spatstat.data_2.1-0       survival_3.2-13          
##  [37] zoo_1.8-9                 glue_1.4.2               
##  [39] polyclip_1.10-0           gtable_0.3.0             
##  [41] zlibbioc_1.38.0           XVector_0.32.0           
##  [43] leiden_0.3.9              DelayedArray_0.18.0      
##  [45] BiocSingular_1.8.1        future.apply_1.8.1       
##  [47] abind_1.4-5               scales_1.1.1             
##  [49] pheatmap_1.0.12           DBI_1.1.1                
##  [51] miniUI_0.1.1.1            Rcpp_1.0.7               
##  [53] viridisLite_0.4.0         xtable_1.8-4             
##  [55] reticulate_1.20           spatstat.core_2.3-0      
##  [57] rsvd_1.0.5                bit_4.0.4                
##  [59] htmlwidgets_1.5.4         httr_1.4.2               
##  [61] FNN_1.1.3                 RColorBrewer_1.1-2       
##  [63] ellipsis_0.3.2            ica_1.0-2                
##  [65] farver_2.1.0              pkgconfig_2.0.3          
##  [67] sass_0.4.0                uwot_0.1.10              
##  [69] deldir_0.2-10             utf8_1.2.2               
##  [71] labeling_0.4.2            tidyselect_1.1.1         
##  [73] rlang_0.4.11              reshape2_1.4.4           
##  [75] later_1.3.0               munsell_0.5.0            
##  [77] tools_4.1.0               cachem_1.0.6             
##  [79] cli_3.0.1                 generics_0.1.0           
##  [81] RSQLite_2.2.8             ggridges_0.5.3           
##  [83] evaluate_0.14             stringr_1.4.0            
##  [85] fastmap_1.1.0             yaml_2.2.1               
##  [87] goftest_1.2-2             knitr_1.34               
##  [89] bit64_4.0.5               fitdistrplus_1.1-5       
##  [91] DDRTree_0.1.5             purrr_0.3.4              
##  [93] RANN_2.6.1                KEGGREST_1.32.0          
##  [95] sparseMatrixStats_1.4.2   pbapply_1.4-3            
##  [97] future_1.22.1             nlme_3.1-152             
##  [99] mime_0.11                 slam_0.1-48              
## [101] monocle_2.20.0            compiler_4.1.0           
## [103] rstudioapi_0.13           plotly_4.9.4.1           
## [105] png_0.1-7                 spatstat.utils_2.2-0     
## [107] tibble_3.1.4              bslib_0.3.0              
## [109] stringi_1.7.4             highr_0.9                
## [111] RSpectra_0.16-0           lattice_0.20-44          
## [113] Matrix_1.3-4              HSMMSingleCell_1.12.0    
## [115] vctrs_0.3.8               pillar_1.6.2             
## [117] lifecycle_1.0.0           combinat_0.0-8           
## [119] spatstat.geom_2.2-2       lmtest_0.9-38            
## [121] jquerylib_0.1.4           BiocNeighbors_1.10.0     
## [123] RcppAnnoy_0.0.19          data.table_1.14.0        
## [125] cowplot_1.1.1             bitops_1.0-7             
## [127] irlba_2.3.3               httpuv_1.6.3             
## [129] patchwork_1.1.1           R6_2.5.1                 
## [131] bookdown_0.24             promises_1.2.0.1         
## [133] KernSmooth_2.23-20        gridExtra_2.3            
## [135] parallelly_1.28.1         codetools_0.2-18         
## [137] MASS_7.3-54               assertthat_0.2.1         
## [139] withr_2.4.2               qlcMatrix_0.9.7          
## [141] sctransform_0.3.2         GenomeInfoDbData_1.2.6   
## [143] mgcv_1.8-36               beachmat_2.8.1           
## [145] grid_4.1.0                rpart_4.1-15             
## [147] tidyr_1.1.3               DelayedMatrixStats_1.14.3
## [149] rmarkdown_2.10            Rtsne_0.15               
## [151] shiny_1.6.0</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="quantification-1.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/05-protocol.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["scRNA-seq tutorial.pdf", "scRNA-seq tutorial.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
