<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Introduction | scRNA-seq data analysis</title>
  <meta name="description" content="This tutorial contains steps for analysing scRNA-seq data." />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Introduction | scRNA-seq data analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This tutorial contains steps for analysing scRNA-seq data." />
  <meta name="github-repo" content="chisangad/scRNAseq" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Introduction | scRNA-seq data analysis" />
  
  <meta name="twitter:description" content="This tutorial contains steps for analysing scRNA-seq data." />
  

<meta name="author" content="David Chisanga and Wei Shi" />


<meta name="date" content="2021-09-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="prerequisites.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://www.onjcri.org.au/our-research/#bcgl">BCG Lab @ONJCRI</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="intro.html"><a href="intro.html#quantification"><i class="fa fa-check"></i><b>2.1</b> Quantification</a></li>
<li class="chapter" data-level="2.2" data-path="intro.html"><a href="intro.html#quality-control-and-normalization"><i class="fa fa-check"></i><b>2.2</b> Quality control and Normalization</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="intro.html"><a href="intro.html#quality-control"><i class="fa fa-check"></i><b>2.2.1</b> Quality control</a></li>
<li class="chapter" data-level="2.2.2" data-path="intro.html"><a href="intro.html#normalisation-and-data-correction"><i class="fa fa-check"></i><b>2.2.2</b> Normalisation and data correction</a></li>
<li class="chapter" data-level="2.2.3" data-path="intro.html"><a href="intro.html#dimension-reduction"><i class="fa fa-check"></i><b>2.2.3</b> Dimension reduction</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="intro.html"><a href="intro.html#data-analysis"><i class="fa fa-check"></i><b>2.3</b> Data analysis</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>3</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="3.1" data-path="prerequisites.html"><a href="prerequisites.html#prereq"><i class="fa fa-check"></i><b>3.1</b> Data</a></li>
<li class="chapter" data-level="3.2" data-path="prerequisites.html"><a href="prerequisites.html#software"><i class="fa fa-check"></i><b>3.2</b> SOFTWARE</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="running-environment.html"><a href="running-environment.html"><i class="fa fa-check"></i><b>4</b> Running Environment</a>
<ul>
<li class="chapter" data-level="4.1" data-path="running-environment.html"><a href="running-environment.html#rstudio"><i class="fa fa-check"></i><b>4.1</b> RStudio</a></li>
<li class="chapter" data-level="4.2" data-path="running-environment.html"><a href="running-environment.html#setup"><i class="fa fa-check"></i><b>4.2</b> UNIX server + laptop</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quantification-1.html"><a href="quantification-1.html"><i class="fa fa-check"></i><b>5</b> Quantification</a>
<ul>
<li class="chapter" data-level="5.1" data-path="quantification-1.html"><a href="quantification-1.html#build-index-for-a-reference-genome"><i class="fa fa-check"></i><b>5.1</b> Build index for a reference genome</a></li>
<li class="chapter" data-level="5.2" data-path="quantification-1.html"><a href="quantification-1.html#counts"><i class="fa fa-check"></i><b>5.2</b> Map and quantify single cell RNA-seq data</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="analysis-protocol.html"><a href="analysis-protocol.html"><i class="fa fa-check"></i><b>6</b> Analysis protocol</a>
<ul>
<li class="chapter" data-level="6.1" data-path="analysis-protocol.html"><a href="analysis-protocol.html#read-mapping-and-umi-counts"><i class="fa fa-check"></i><b>6.1</b> Read mapping and UMI counts</a></li>
<li class="chapter" data-level="6.2" data-path="analysis-protocol.html"><a href="analysis-protocol.html#quality-control-checks"><i class="fa fa-check"></i><b>6.2</b> Quality control checks</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="analysis-protocol.html"><a href="analysis-protocol.html#mitochondria-content"><i class="fa fa-check"></i><b>6.2.1</b> Mitochondria content</a></li>
<li class="chapter" data-level="6.2.2" data-path="analysis-protocol.html"><a href="analysis-protocol.html#ribosomal-content"><i class="fa fa-check"></i><b>6.2.2</b> Ribosomal content</a></li>
<li class="chapter" data-level="6.2.3" data-path="analysis-protocol.html"><a href="analysis-protocol.html#number-of-detected-genes"><i class="fa fa-check"></i><b>6.2.3</b> Number of detected genes</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="analysis-protocol.html"><a href="analysis-protocol.html#filtering"><i class="fa fa-check"></i><b>6.3</b> Filtering</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="analysis-protocol.html"><a href="analysis-protocol.html#cell-filtering"><i class="fa fa-check"></i><b>6.3.1</b> Cell filtering</a></li>
<li class="chapter" data-level="6.3.2" data-path="analysis-protocol.html"><a href="analysis-protocol.html#gene-filtering"><i class="fa fa-check"></i><b>6.3.2</b> Gene filtering</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="analysis-protocol.html"><a href="analysis-protocol.html#normalization"><i class="fa fa-check"></i><b>6.4</b> Normalization</a></li>
<li class="chapter" data-level="6.5" data-path="analysis-protocol.html"><a href="analysis-protocol.html#dimension-reduction-and-clustering"><i class="fa fa-check"></i><b>6.5</b> Dimension reduction and clustering</a></li>
<li class="chapter" data-level="6.6" data-path="analysis-protocol.html"><a href="analysis-protocol.html#cell-annotation"><i class="fa fa-check"></i><b>6.6</b> Cell annotation</a></li>
<li class="chapter" data-level="6.7" data-path="analysis-protocol.html"><a href="analysis-protocol.html#trajectory-analysis"><i class="fa fa-check"></i><b>6.7</b> Trajectory analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">scRNA-seq data analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="intro" class="section level1" number="2">
<h1><span class="header-section-number">Chapter 2</span> Introduction</h1>
<p>Single-cell RNA-seq (scRNA-seq) first published by <span class="citation">(<a href="#ref-tang2009mrna" role="doc-biblioref">Tang et al. 2009</a>)</span> measures the distribution of expression levels in individual cells. This allows for the study of transcriptome changes at the cellular level to provide unprecedented resolution of gene expression changes in complex cellular systems. As in bulk RNA-seq, scRNA-seq involves a range of analyses including quantification, dimension reduction, cell clustering, cell type identification, inference of gene-regulatory networks, differential expression and cell hierarchy reconstruction. This scRNA-seq protocol provides a step-by-step overview of a standard scRNA-seq analysis. The steps can be categorized into 3 groups; quantification, quality control and downstream analysis</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:fig1"></span>
<img src="Fig1.png" alt="Overview of the scRNA-seq analysis protocol" width="960" />
<p class="caption">
Figure 2.1: Overview of the scRNA-seq analysis protocol
</p>
</div>
<div id="quantification" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Quantification</h2>
<p>Converting the raw data into a matrix of counts is the first step in a scRNA-seq analysis. <em>cellCounts</em> a scRNA-seq quantification program within <strong>Rsubread</strong> is used in the quantification step. <a href="https://rdrr.io/bioc/Rsubread/man/cellCounts.html"><strong>cellCounts</strong></a> is a program developed for quantifying single-cell RNA-seq (scRNA-seq) data generated from the <a href="https://www.10xgenomics.com/products/single-cell-gene-expression">10X platform</a>. cellCounts takes as input scRNA-seq reads, maps them to the reference genome and then produces UMI (Unique Molecular Identifier) counts for each gene in each cell. Its read mapping is based on the <strong>align</strong> program <span class="citation">(<a href="#ref-Rsubread2019" role="doc-biblioref">Liao, Smyth, and Shi 2019</a>; <a href="#ref-liao2013subread" role="doc-biblioref">Liao, Smyth, and Shi 2013b</a>)</span> and its UMI counting is based on the <em>featureCounts</em> program <span class="citation">(<a href="#ref-liao2013featurecounts" role="doc-biblioref">Liao, Smyth, and Shi 2013a</a>)</span>. Sample demultiplexing, cell barcode demultiplexing and read deduplication are carried out before generating the UMI counts.</p>
<p>The <em>cellCounts</em> function returns a R list object which contains the following components:</p>
<p><strong>counts</strong><br />
A List object including UMI counts for each sample. Each component in this object is a matrix that contains UMI counts for a sample. Rows in the matrix are genes and columns are cells.</p>
<p><strong>annotation</strong><br />
A data.frame object containing a gene annotation. This is the annotation that was used for the assignment of UMIs to genes during quantification. Rows in the annotation are genes. Columns of the annotation include GeneID, Chr, Start, End and Length.</p>
<p><strong>sample.info</strong><br />
A data.frame object containing sample information and quantification statistics. It includes the following columns: SampleName, InputDirectory (if the input format is BCL), TotalCells,HighConfidenceCells (if umi.cutoff is NULL), RescuedCells (if umi.cutoff is NULL), TotalUMI, MinUMI, MedianUMI, MaxUMI, MeanUMI, TotalReads, MappedReads and AssignedReads. Each row in the data frame is a sample.</p>
<p><strong>cell.confidence</strong><br />
A List object indicating if a cell is a high-confidence cell or a rescued cell (low confidence). Each component in the object is a logical vector indicating which cells in a sample are high-confidence cells. cell confidence is included in the output only if umi cutoff is NULL.</p>
</div>
<div id="quality-control-and-normalization" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Quality control and Normalization</h2>
<p>This step largely consists of quality control, normalization, data correction and dimension reduction. For this step, we will use visualization plots and utilize a number of functions within the <strong>Seurat</strong> <span class="citation">(<a href="#ref-hao2021" role="doc-biblioref">Hao et al. 2021</a>)</span> pipeline for scRNA-seq analysis.</p>
<div id="quality-control" class="section level3" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> Quality control</h3>
<p><strong>Cell filtering</strong><br />
</p>
<p>Quality control is performed in several ways to remove;</p>
<ul>
<li>low quality cells</li>
<li>contaminants</li>
</ul>
<p>A number of metrics are used to assess the quality of the scRNA-seq data including the proportion of detected genes in each cell, and fraction of UMI accounts associated with Mitochondrial/Ribosomal genes.</p>
<p>Cells with a low or high number of detected genes are indicative of</p>
<ul>
<li>quiescent/damaged cells or</li>
<li>doublets/multiplets <span class="citation">(<a href="#ref-nayak2021hitchhiker" role="doc-biblioref">Nayak and Hasija 2021</a>)</span></li>
</ul>
<p>A high fraction of mitochondrial genes in a cell is associated with <a href="https://kb.10xgenomics.com/hc/en-us/articles/360001086611-Why-do-I-see-a-high-level-of-mitochondrial-gene-expression-">(See 10X website)</a></p>
<ul>
<li>poor sample quality which leads to a high fraction of apoptotic or lysing cells.</li>
<li>biology of the particular sample, for example, tumor biopsies, may have increased mitochondrial gene expression due to metabolic activity and/or necrosis.</li>
</ul>
<p>A high fraction of ribosomal genes in each cell is indicative of RNA degradation which leads to more templating of rRNA-fragments <a href="https://kb.10xgenomics.com/hc/en-us/articles/218169723-What-fraction-of-reads-map-to-ribosomal-proteins-">(See 10X website)</a></p>
<p>To decide what cut-offs to use, we will visualize the distributions of each metric using plots such as histograms and box plots.</p>
<p><strong>Gene filtering</strong><br />
</p>
<p>Genes not meeting the following criteria are also filtered;</p>
<ul>
<li>Genes not expressed in at least a given number of cells</li>
<li>Genes that may have very high expression</li>
<li>Mitochondrial encoded genes</li>
<li>Ribosomal encoded genes</li>
<li>Genes deemed to induce technical bias</li>
</ul>
</div>
<div id="normalisation-and-data-correction" class="section level3" number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> Normalisation and data correction</h3>
<p>The filtered count matrix is normalized to account for gene expression variability between cells. There are several normalization methods, a number of which have been derived from Bulk RNA-seq and include;</p>
<ul>
<li>CPM (Counts Per Million)</li>
<li>TPM (Transcripts Per Million)</li>
<li>FPKM (Fragments Per Kilobase Million)</li>
<li>TMM (Trimmed Mean of M-values)</li>
<li>Quantile normalization</li>
</ul>
<p>The sparsity and high heterogeneity in scRNA-seq adds to the complexity of normalizing scRNA-seq data. Here, we will utilize normalization methods included in the <strong>Seurat</strong> package. We will be using a global-scaling normalization method “LogNormalize” in <strong>Seurat</strong>, this method normalizes the feature expression measurements for each cell by the total expression, multiplies this by a scale factor (10,000 by default), and log-transforms the result. In addition, for Differential Expression analysis, we utilize the the limma-voom pipeline <span class="citation">(<a href="#ref-ritchie2015limma" role="doc-biblioref">Ritchie et al. 2015, law2014voom</a>)</span> and here, we will be using the CPM normalization method.</p>
</div>
<div id="dimension-reduction" class="section level3" number="2.2.3">
<h3><span class="header-section-number">2.2.3</span> Dimension reduction</h3>
<p>The resulting count matrix from scRNA-seq is high dimensional in nature. Furthermore, not all the genes included in the matrix are needed for the classification of the cellular expression profiles. As such, it is important that we focus on the Biological signals that are informative. Popular techniques for dimension reduction are;</p>
<ul>
<li>PCA (Principal Component Analysis)</li>
<li>t-distributed Stochastic Neighbor Embedding (t-SNE)</li>
<li>Uniform Manifold Approximation, and Projection (UMAP)</li>
<li>Self Organizing Maps (SOM)</li>
</ul>
<p>There are two components involved in dimension reduction;</p>
<ul>
<li>Feature selection where a small subset of features is selected</li>
<li>Feature extraction where higher dimension data is projected to a lower dimension</li>
</ul>
<p>In this workshop, t-SNE and UMAP are used to demonstrate how to group cells into clusters.</p>
</div>
</div>
<div id="data-analysis" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Data analysis</h2>
<p>After the data has been processed, there are several analyses that can be done depending on the research question. In this tutorial, we will look at the standard analysis in scRNA-seq such as Cluster analysis, cell type annotation, and Trajectory inference/pseudo-temporal ordering. We will also perform gene-level analysis including differential expression analysis, pathway and gene set enrichment analysis, and the generation of Gene Regulatory Networks</p>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-hao2021" class="csl-entry">
Hao, Yuhan, Stephanie Hao, Erica Andersen-Nissen, William M. Mauck III, Shiwei Zheng, Andrew Butler, Maddie J. Lee, et al. 2021. <span>“Integrated Analysis of Multimodal Single-Cell Data.”</span> <em>Cell</em>. <a href="https://doi.org/10.1016/j.cell.2021.04.048">https://doi.org/10.1016/j.cell.2021.04.048</a>.
</div>
<div id="ref-Rsubread2019" class="csl-entry">
Liao, Yang, Gordon K. Smyth, and Wei Shi. 2019. <span>“The <span>R</span> Package <span>Rsubread</span> Is Easier, Faster, Cheaper and Better for Alignment and Quantification of <span>RNA</span> Sequencing Reads.”</span> <em>Nucleic Acids Research</em> 47: e47. <a href="https://doi.org/10.1093/nar/gkz114">https://doi.org/10.1093/nar/gkz114</a>.
</div>
<div id="ref-liao2013featurecounts" class="csl-entry">
Liao, Yang, Gordon K Smyth, and Wei Shi. 2013a. <span>“featureCounts: An Efficient General Purpose Program for Assigning Sequence Reads to Genomic Features.”</span> <em>Bioinformatics</em> 30 (7): 923–30.
</div>
<div id="ref-liao2013subread" class="csl-entry">
———. 2013b. <span>“The Subread Aligner: Fast, Accurate and Scalable Read Mapping by Seed-and-Vote.”</span> <em>Nucleic Acids Research</em> 41 (10): e108–8.
</div>
<div id="ref-nayak2021hitchhiker" class="csl-entry">
Nayak, Richa, and Yasha Hasija. 2021. <span>“A Hitchhiker’s Guide to Single-Cell Transcriptomics and Data Analysis Pipelines.”</span> <em>Genomics</em>.
</div>
<div id="ref-ritchie2015limma" class="csl-entry">
Ritchie, Matthew E, Belinda Phipson, Di Wu, Yifang Hu, Charity W Law, Wei Shi, and Gordon K Smyth. 2015. <span>“Limma Powers Differential Expression Analyses for RNA-Sequencing and Microarray Studies.”</span> <em>Nucleic Acids Research</em> 43 (7): e47–47.
</div>
<div id="ref-tang2009mrna" class="csl-entry">
Tang, Fuchou, Catalin Barbacioru, Yangzhou Wang, Ellen Nordman, Clarence Lee, Nanlan Xu, Xiaohui Wang, et al. 2009. <span>“mRNA-Seq Whole-Transcriptome Analysis of a Single Cell.”</span> <em>Nature Methods</em> 6 (5): 377–82.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="prerequisites.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/01-intro.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["scRNA-seq tutorial.pdf", "scRNA-seq tutorial.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
