<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Quantification | scRNA-seq data analysis</title>
  <meta name="description" content="This tutorial contains steps for analysing scRNA-seq data using Rsubread cellCounts + Seurat pipeline." />
  <meta name="generator" content="bookdown 0.27 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Quantification | scRNA-seq data analysis" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This tutorial contains steps for analysing scRNA-seq data using Rsubread cellCounts + Seurat pipeline." />
  <meta name="github-repo" content="chisangad/scRNAseq" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Quantification | scRNA-seq data analysis" />
  
  <meta name="twitter:description" content="This tutorial contains steps for analysing scRNA-seq data using Rsubread cellCounts + Seurat pipeline." />
  

<meta name="author" content="David Chisanga and Wei Shi" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="running-environment.html"/>
<link rel="next" href="analysis-protocol.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://www.onjcri.org.au/our-research/#bcgl">BCG Lab @ONJCRI</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="intro.html"><a href="intro.html#quantification"><i class="fa fa-check"></i><b>2.1</b> Quantification</a></li>
<li class="chapter" data-level="2.2" data-path="intro.html"><a href="intro.html#quality-control-and-normalization"><i class="fa fa-check"></i><b>2.2</b> Quality control and Normalization</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="intro.html"><a href="intro.html#quality-control"><i class="fa fa-check"></i><b>2.2.1</b> Quality control</a></li>
<li class="chapter" data-level="2.2.2" data-path="intro.html"><a href="intro.html#normalisation-and-data-correction"><i class="fa fa-check"></i><b>2.2.2</b> Normalisation and data correction</a></li>
<li class="chapter" data-level="2.2.3" data-path="intro.html"><a href="intro.html#dimension-reduction"><i class="fa fa-check"></i><b>2.2.3</b> Dimension reduction</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="intro.html"><a href="intro.html#data-analysis"><i class="fa fa-check"></i><b>2.3</b> Data analysis</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>3</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="3.1" data-path="prerequisites.html"><a href="prerequisites.html#prereq"><i class="fa fa-check"></i><b>3.1</b> Data</a></li>
<li class="chapter" data-level="3.2" data-path="prerequisites.html"><a href="prerequisites.html#software"><i class="fa fa-check"></i><b>3.2</b> SOFTWARE</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="running-environment.html"><a href="running-environment.html"><i class="fa fa-check"></i><b>4</b> Running Environment</a>
<ul>
<li class="chapter" data-level="4.1" data-path="running-environment.html"><a href="running-environment.html#rstudio"><i class="fa fa-check"></i><b>4.1</b> RStudio</a></li>
<li class="chapter" data-level="4.2" data-path="running-environment.html"><a href="running-environment.html#setup"><i class="fa fa-check"></i><b>4.2</b> UNIX server + laptop</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quantification-1.html"><a href="quantification-1.html"><i class="fa fa-check"></i><b>5</b> Quantification</a>
<ul>
<li class="chapter" data-level="5.1" data-path="quantification-1.html"><a href="quantification-1.html#build-index-for-a-reference-genome"><i class="fa fa-check"></i><b>5.1</b> Build index for a reference genome</a></li>
<li class="chapter" data-level="5.2" data-path="quantification-1.html"><a href="quantification-1.html#counts"><i class="fa fa-check"></i><b>5.2</b> Map and quantify single cell RNA-seq data</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="analysis-protocol.html"><a href="analysis-protocol.html"><i class="fa fa-check"></i><b>6</b> Analysis protocol</a>
<ul>
<li class="chapter" data-level="6.1" data-path="analysis-protocol.html"><a href="analysis-protocol.html#read-mapping-and-umi-counts"><i class="fa fa-check"></i><b>6.1</b> Read mapping and UMI counts</a></li>
<li class="chapter" data-level="6.2" data-path="analysis-protocol.html"><a href="analysis-protocol.html#quality-control-checks"><i class="fa fa-check"></i><b>6.2</b> Quality control checks</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="analysis-protocol.html"><a href="analysis-protocol.html#mitochondria-content"><i class="fa fa-check"></i><b>6.2.1</b> Mitochondria content</a></li>
<li class="chapter" data-level="6.2.2" data-path="analysis-protocol.html"><a href="analysis-protocol.html#ribosomal-content"><i class="fa fa-check"></i><b>6.2.2</b> Ribosomal content</a></li>
<li class="chapter" data-level="6.2.3" data-path="analysis-protocol.html"><a href="analysis-protocol.html#number-of-detected-genes"><i class="fa fa-check"></i><b>6.2.3</b> Number of detected genes</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="analysis-protocol.html"><a href="analysis-protocol.html#filtering"><i class="fa fa-check"></i><b>6.3</b> Filtering</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="analysis-protocol.html"><a href="analysis-protocol.html#cell-filtering"><i class="fa fa-check"></i><b>6.3.1</b> Cell filtering</a></li>
<li class="chapter" data-level="6.3.2" data-path="analysis-protocol.html"><a href="analysis-protocol.html#gene-filtering"><i class="fa fa-check"></i><b>6.3.2</b> Gene filtering</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="analysis-protocol.html"><a href="analysis-protocol.html#normalization"><i class="fa fa-check"></i><b>6.4</b> Normalization</a></li>
<li class="chapter" data-level="6.5" data-path="analysis-protocol.html"><a href="analysis-protocol.html#dimension-reduction-and-clustering"><i class="fa fa-check"></i><b>6.5</b> Dimension reduction and clustering</a></li>
<li class="chapter" data-level="6.6" data-path="analysis-protocol.html"><a href="analysis-protocol.html#cell-annotation"><i class="fa fa-check"></i><b>6.6</b> Cell annotation</a></li>
<li class="chapter" data-level="6.7" data-path="analysis-protocol.html"><a href="analysis-protocol.html#trajectory-analysis"><i class="fa fa-check"></i><b>6.7</b> Trajectory analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">scRNA-seq data analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="quantification-1" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Quantification<a href="quantification-1.html#quantification-1" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="build-index-for-a-reference-genome" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Build index for a reference genome<a href="quantification-1.html#build-index-for-a-reference-genome" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Start an R session using the terminal on the server</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="quantification-1.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load R</span>
<span id="cb9-2"><a href="quantification-1.html#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="ex">R</span></span></code></pre></div>
<p>First change to the working directory ‘Workshop_scRNAseq’ using the command below, remember to replace ‘my_directory’ with the name of the directory you created in <a href="running-environment.html#setup">4.2</a>.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="quantification-1.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/data/Processing/Public/my_directory/Workshop_scRNAseq/&quot;</span>)</span></code></pre></div>
<p>Build an index for the reference genome <em>GRCh38/hg38</em> using the <em>buildindex</em> function in <strong>Rsubread</strong>. The created index files will be saved to the current working directory. This index only needs to be built once and it can be reused in future RNA-seq (both bulk and single) data analyses.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="quantification-1.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Rsubread)</span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="quantification-1.html#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">buildindex</span>(<span class="at">basename =</span> <span class="st">&quot;hg38_reference&quot;</span>, </span>
<span id="cb12-2"><a href="quantification-1.html#cb12-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">reference =</span> <span class="st">&quot;GRCh38.primary_assembly.genome.fa.gz&quot;</span>)</span></code></pre></div>
</div>
<div id="counts" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Map and quantify single cell RNA-seq data<a href="quantification-1.html#counts" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The raw scRNA-seq data is processed and UMI counts generated using <em>cellCounts</em>. It takes as input scRNA-seq reads generated by the 10X platform, maps them to the reference genome and then produces UMI (Unique Molecular Identifier) counts for each gene in each cell. It uses the <em>align</em> read mapping function and the <em>featureCounts</em> quantification function, both part of the <strong>Rsubread</strong> package.</p>
<p>For more information on how to use <em>cellCounts</em>, type the command <code>?cellCounts</code> in the R console.</p>
<p><strong>Create sample-related information</strong></p>
<p>Before using cellCounts, we need to generate a data frame containing sample information which we then pass to cellCounts via the sample parameter.</p>
<p>Note that for this workshop, <em>the input format for our dataset is FASTQ</em> and as such we create a data frame that consists of 3 columns ‘BarcodeUMIFile’, ‘ReadFile’ and ‘SampleName’. However, if the input format were BCL (ie. input.mode=“BCL”), we would have created a dataframe that included the location where the read data are stored, flowcell lanes used for sequencing, sample names and names of index sets used for indexing samples.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="quantification-1.html#cb13-1" aria-hidden="true" tabindex="-1"></a>umiread <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&quot;FQs&quot;</span>,</span>
<span id="cb13-2"><a href="quantification-1.html#cb13-2" aria-hidden="true" tabindex="-1"></a>                      <span class="at">pattern =</span> <span class="st">&quot;^.*_1.fastq.gz&quot;</span>,</span>
<span id="cb13-3"><a href="quantification-1.html#cb13-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">full.names =</span> T)</span>
<span id="cb13-4"><a href="quantification-1.html#cb13-4" aria-hidden="true" tabindex="-1"></a>readfile <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;1.fastq.gz&quot;</span>, <span class="st">&quot;2.fastq.gz&quot;</span>, umiread)</span>
<span id="cb13-5"><a href="quantification-1.html#cb13-5" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;10X_|_1.fastq.gz&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">basename</span>(umiread))</span>
<span id="cb13-6"><a href="quantification-1.html#cb13-6" aria-hidden="true" tabindex="-1"></a>sample.sheet <span class="ot">&lt;-</span></span>
<span id="cb13-7"><a href="quantification-1.html#cb13-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb13-8"><a href="quantification-1.html#cb13-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">BarcodeUMIFile =</span> umiread,</span>
<span id="cb13-9"><a href="quantification-1.html#cb13-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">ReadFile =</span> readfile,</span>
<span id="cb13-10"><a href="quantification-1.html#cb13-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">SampleName =</span> samples,</span>
<span id="cb13-11"><a href="quantification-1.html#cb13-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">stringsAsFactors =</span> F</span>
<span id="cb13-12"><a href="quantification-1.html#cb13-12" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
<p><strong>Align and count reads</strong></p>
<p>Sequence reads are mapped to the human genome (GRCh38/hg38) and UMIs are counted to each gene using the inbuilt human (<em>hg38</em>) annotation. The <em>cellCounts</em> function returns a list object to R. It also outputs one BAM file for each sample which includes location-sorted read mapping results. The returned R list object is saved as an R object “counts.RData”, this will then be loaded to perform further downstream analysis.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="quantification-1.html#cb14-1" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span></span>
<span id="cb14-2"><a href="quantification-1.html#cb14-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cellCounts</span>(</span>
<span id="cb14-3"><a href="quantification-1.html#cb14-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">index =</span> <span class="st">&quot;hg38_reference&quot;</span>,</span>
<span id="cb14-4"><a href="quantification-1.html#cb14-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">sample =</span> sample.sheet,</span>
<span id="cb14-5"><a href="quantification-1.html#cb14-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">annot.inbuilt =</span> <span class="st">&quot;hg38&quot;</span>,</span>
<span id="cb14-6"><a href="quantification-1.html#cb14-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">input.mode =</span> <span class="st">&quot;FASTQ&quot;</span>,</span>
<span id="cb14-7"><a href="quantification-1.html#cb14-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">nthreads =</span> <span class="dv">20</span></span>
<span id="cb14-8"><a href="quantification-1.html#cb14-8" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb14-9"><a href="quantification-1.html#cb14-9" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(counts, <span class="at">file =</span> <span class="st">&quot;counts.RData&quot;</span>)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="running-environment.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="analysis-protocol.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/04-quantification.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["scRNAseqtutorial.pdf", "scRNAseqtutorial.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
