<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Running Environment | scRNA-seq data analysis</title>
  <meta name="description" content="This tutorial contains steps for analysing scRNA-seq data." />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Running Environment | scRNA-seq data analysis" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This tutorial contains steps for analysing scRNA-seq data." />
  <meta name="github-repo" content="chisangad/scRNAseq" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Running Environment | scRNA-seq data analysis" />
  
  <meta name="twitter:description" content="This tutorial contains steps for analysing scRNA-seq data." />
  

<meta name="author" content="David Chisanga and Wei Shi" />


<meta name="date" content="2021-09-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="prerequisites.html"/>
<link rel="next" href="quantification-1.html"/>
<script src="libs/header-attrs-2.10/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="https://www.onjcri.org.au/our-research/#bcgl">BCG Lab @ONJCRI</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="intro.html"><a href="intro.html#quantification"><i class="fa fa-check"></i><b>2.1</b> Quantification</a></li>
<li class="chapter" data-level="2.2" data-path="intro.html"><a href="intro.html#quality-control-and-normalization"><i class="fa fa-check"></i><b>2.2</b> Quality control and Normalization</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="intro.html"><a href="intro.html#quality-control"><i class="fa fa-check"></i><b>2.2.1</b> Quality control</a></li>
<li class="chapter" data-level="2.2.2" data-path="intro.html"><a href="intro.html#normalisation-and-data-correction"><i class="fa fa-check"></i><b>2.2.2</b> Normalisation and data correction</a></li>
<li class="chapter" data-level="2.2.3" data-path="intro.html"><a href="intro.html#dimension-reduction"><i class="fa fa-check"></i><b>2.2.3</b> Dimension reduction</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="intro.html"><a href="intro.html#data-analysis"><i class="fa fa-check"></i><b>2.3</b> Data analysis</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="prerequisites.html"><a href="prerequisites.html"><i class="fa fa-check"></i><b>3</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="3.1" data-path="prerequisites.html"><a href="prerequisites.html#prereq"><i class="fa fa-check"></i><b>3.1</b> Data</a></li>
<li class="chapter" data-level="3.2" data-path="prerequisites.html"><a href="prerequisites.html#software"><i class="fa fa-check"></i><b>3.2</b> SOFTWARE</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="running-environment.html"><a href="running-environment.html"><i class="fa fa-check"></i><b>4</b> Running Environment</a>
<ul>
<li class="chapter" data-level="4.1" data-path="running-environment.html"><a href="running-environment.html#rstudio"><i class="fa fa-check"></i><b>4.1</b> RStudio</a></li>
<li class="chapter" data-level="4.2" data-path="running-environment.html"><a href="running-environment.html#setup"><i class="fa fa-check"></i><b>4.2</b> UNIX server + laptop</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="quantification-1.html"><a href="quantification-1.html"><i class="fa fa-check"></i><b>5</b> Quantification</a>
<ul>
<li class="chapter" data-level="5.1" data-path="quantification-1.html"><a href="quantification-1.html#build-index-for-a-reference-genome"><i class="fa fa-check"></i><b>5.1</b> Build index for a reference genome</a></li>
<li class="chapter" data-level="5.2" data-path="quantification-1.html"><a href="quantification-1.html#counts"><i class="fa fa-check"></i><b>5.2</b> Map and quantify single cell RNA-seq data</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="analysis-protocol.html"><a href="analysis-protocol.html"><i class="fa fa-check"></i><b>6</b> Analysis protocol</a>
<ul>
<li class="chapter" data-level="6.1" data-path="analysis-protocol.html"><a href="analysis-protocol.html#read-mapping-and-umi-counts"><i class="fa fa-check"></i><b>6.1</b> Read mapping and UMI counts</a></li>
<li class="chapter" data-level="6.2" data-path="analysis-protocol.html"><a href="analysis-protocol.html#quality-control-checks"><i class="fa fa-check"></i><b>6.2</b> Quality control checks</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="analysis-protocol.html"><a href="analysis-protocol.html#mitochondria-content"><i class="fa fa-check"></i><b>6.2.1</b> Mitochondria content</a></li>
<li class="chapter" data-level="6.2.2" data-path="analysis-protocol.html"><a href="analysis-protocol.html#ribosomal-content"><i class="fa fa-check"></i><b>6.2.2</b> Ribosomal content</a></li>
<li class="chapter" data-level="6.2.3" data-path="analysis-protocol.html"><a href="analysis-protocol.html#number-of-detected-genes"><i class="fa fa-check"></i><b>6.2.3</b> Number of detected genes</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="analysis-protocol.html"><a href="analysis-protocol.html#filtering"><i class="fa fa-check"></i><b>6.3</b> Filtering</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="analysis-protocol.html"><a href="analysis-protocol.html#cell-filtering"><i class="fa fa-check"></i><b>6.3.1</b> Cell filtering</a></li>
<li class="chapter" data-level="6.3.2" data-path="analysis-protocol.html"><a href="analysis-protocol.html#gene-filtering"><i class="fa fa-check"></i><b>6.3.2</b> Gene filtering</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="analysis-protocol.html"><a href="analysis-protocol.html#normalization"><i class="fa fa-check"></i><b>6.4</b> Normalization</a></li>
<li class="chapter" data-level="6.5" data-path="analysis-protocol.html"><a href="analysis-protocol.html#dimension-reduction-and-clustering"><i class="fa fa-check"></i><b>6.5</b> Dimension reduction and clustering</a></li>
<li class="chapter" data-level="6.6" data-path="analysis-protocol.html"><a href="analysis-protocol.html#cell-annotation"><i class="fa fa-check"></i><b>6.6</b> Cell annotation</a></li>
<li class="chapter" data-level="6.7" data-path="analysis-protocol.html"><a href="analysis-protocol.html#trajectory-analysis"><i class="fa fa-check"></i><b>6.7</b> Trajectory analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">scRNA-seq data analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="running-environment" class="section level1" number="4">
<h1><span class="header-section-number">Chapter 4</span> Running Environment</h1>
<div id="rstudio" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> RStudio</h2>
<p>This is the recommended approach for running this protocol. An RStudio application (‘RStudio(2672)’) has been created under the ‘Analysis’ tab in the Remote Access Facility. After you log into it (using your email account name and password), change to the directory ‘/data/RawPrimary/Public’ (you can do this by choosing Session &gt; Set Working Directory &gt;Choose Directory and then using the ‘…’ to specify the directory you want to change to) and then
you will find a folder called ‘Workshop_scRNAseq’ which includes all the materials included in this Workshop. You can make a copy of this folder and then start to run the protocol in your own folder.</p>
</div>
<div id="setup" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> UNIX server + laptop</h2>
<p>Run read mapping and counting on a UNIX server and then perform the rest of the analysis on your laptop. Refer to the document “How to Access Linux BioInformatics Analysis Platform.pdf” for how to access the Bioinformatics UNIX server. Once you logged in onto the server, you can issue the following commands to copy the Workshop data to a directory you create on the server and to launch R. You are then ready to run the protocol.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="running-environment.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># connect AllStaffShare Drive to the server</span></span>
<span id="cb2-2"><a href="running-environment.html#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="ex">cifscreds</span> add svr-fs95</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="running-environment.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># change directory to the Public directory</span></span>
<span id="cb3-2"><a href="running-environment.html#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> /data/Processing/Public/</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="running-environment.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create your own directory, eg ‘my_directory’</span></span>
<span id="cb4-2"><a href="running-environment.html#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> my_directory</span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="running-environment.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># copy the Workshop data to your directory</span></span>
<span id="cb5-2"><a href="running-environment.html#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> <span class="at">-r</span> /mnt/AllStaffShare/Workshop_scRNAseq my_directory</span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="running-environment.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Change directory to the Workshop_scRNAseq materials</span></span>
<span id="cb6-2"><a href="running-environment.html#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> my_directory/Workshop_scRNAseq</span></code></pre></div>
<p><strong>R</strong> has already been installed for you, use the <em>module load R</em> command to activate <strong>R</strong></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="running-environment.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># activate R environment</span></span>
<span id="cb7-2"><a href="running-environment.html#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="ex">module</span> load R</span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="running-environment.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Start R environment</span></span>
<span id="cb8-2"><a href="running-environment.html#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="ex">R</span></span></code></pre></div>
<p>Due to the complexity and resource requirements of scRNA-seq data analysis, the quantification step will be run on the server and the rest of the analysis can be run on your laptop.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="prerequisites.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="quantification-1.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/03-environment.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["scRNA-seq tutorial.pdf", "scRNA-seq tutorial.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
